<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_101) on Mon Aug 01 11:40:38 CDT 2016 -->
<title>Index</title>
<meta name="date" content="2016-08-01">
<link rel="stylesheet" type="text/css" href="stylesheet.css" title="Style">
<script type="text/javascript" src="script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Index";
        }
    }
    catch(err) {
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="overview-summary.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li><a href="overview-tree.html">Tree</a></li>
<li><a href="deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="index.html?index-all.html" target="_top">Frames</a></li>
<li><a href="index-all.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="contentContainer"><a href="#I:A">A</a>&nbsp;<a href="#I:B">B</a>&nbsp;<a href="#I:C">C</a>&nbsp;<a href="#I:D">D</a>&nbsp;<a href="#I:E">E</a>&nbsp;<a href="#I:F">F</a>&nbsp;<a href="#I:G">G</a>&nbsp;<a href="#I:H">H</a>&nbsp;<a href="#I:I">I</a>&nbsp;<a href="#I:L">L</a>&nbsp;<a href="#I:M">M</a>&nbsp;<a href="#I:O">O</a>&nbsp;<a href="#I:P">P</a>&nbsp;<a href="#I:S">S</a>&nbsp;<a href="#I:T">T</a>&nbsp;<a href="#I:U">U</a>&nbsp;<a href="#I:V">V</a>&nbsp;<a href="#I:W">W</a>&nbsp;<a name="I:A">
<!--   -->
</a>
<h2 class="title">A</h2>
<dl>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.RecordFilter.html#acceptRecord-edu.uab.ssg.io.ncbi.SeqGeneMdParser.Record-">acceptRecord(SeqGeneMdParser.Record)</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.RecordFilter.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.RecordFilter</a></dt>
<dd>
<div class="block">Returns true if the given record meets the filter criteria.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/AlleleCounter.html#addAllele-java.lang.String-">addAllele(String)</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/AlleleCounter.html" title="class in edu.uab.ssg.model.snp">AlleleCounter</a></dt>
<dd>
<div class="block">Adds an allele to the count.</div>
</dd>
<dt><a href="edu/uab/ssg/model/snp/AlleleCounter.html" title="class in edu.uab.ssg.model.snp"><span class="typeNameLink">AlleleCounter</span></a> - Class in <a href="edu/uab/ssg/model/snp/package-summary.html">edu.uab.ssg.model.snp</a></dt>
<dd>
<div class="block">A helper class for counting alleles and reporting their frequencies,
 the name of the minor allele (if it exists), the number of missing
 values, as well as whether the marker appears to be biallelic, 
 monomorphic, or from an ambiguous A/T or G/C SNP.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/AlleleCounter.html#AlleleCounter--">AlleleCounter()</a></span> - Constructor for class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/AlleleCounter.html" title="class in edu.uab.ssg.model.snp">AlleleCounter</a></dt>
<dd>
<div class="block">Constructs an allele counter.</div>
</dd>
<dt><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.html" title="class in edu.uab.ssg.io.hapmap"><span class="typeNameLink">AlleleFreqFileParser</span></a> - Class in <a href="edu/uab/ssg/io/hapmap/package-summary.html">edu.uab.ssg.io.hapmap</a></dt>
<dd>
<div class="block">A parser for the allele frequency file format for the HapMap phase II data located at <a href="http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/latest_phaseII_ncbi_b36/">http://hapmap.ncbi.nlm.nih.gov/downloads/frequencies/latest_phaseII_ncbi_b36/</a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.html#AlleleFreqFileParser--">AlleleFreqFileParser()</a></span> - Constructor for class edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.html" title="class in edu.uab.ssg.io.hapmap">AlleleFreqFileParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap"><span class="typeNameLink">AlleleFreqFileParser.AlleleFreqRecord</span></a> - Interface in <a href="edu/uab/ssg/io/hapmap/package-summary.html">edu.uab.ssg.io.hapmap</a></dt>
<dd>
<div class="block">An allele frequency record.</div>
</dd>
<dt><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.hapmap"><span class="typeNameLink">AlleleFreqFileParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/hapmap/package-summary.html">edu.uab.ssg.io.hapmap</a></dt>
<dd>
<div class="block">A listener for handling parsed allele frequency records and problems
 due to bad record formatting.</div>
</dd>
</dl>
<a name="I:B">
<!--   -->
</a>
<h2 class="title">B</h2>
<dl>
<dt><a href="edu/uab/ssg/io/plink/BIMParser.html" title="class in edu.uab.ssg.io.plink"><span class="typeNameLink">BIMParser</span></a> - Class in <a href="edu/uab/ssg/io/plink/package-summary.html">edu.uab.ssg.io.plink</a></dt>
<dd>
<div class="block">A parser for the PLINK extended MAP input file format described at <a href="http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#bed">http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#bed</a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/BIMParser.html#BIMParser--">BIMParser()</a></span> - Constructor for class edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/BIMParser.html" title="class in edu.uab.ssg.io.plink">BIMParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/plink/BIMParser.BadRecordFormatListener.html" title="interface in edu.uab.ssg.io.plink"><span class="typeNameLink">BIMParser.BadRecordFormatListener</span></a> - Interface in <a href="edu/uab/ssg/io/plink/package-summary.html">edu.uab.ssg.io.plink</a></dt>
<dd>
<div class="block">A listener for handling problems due to bad record formatting.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/IlluminaStrand.html#BOT">BOT</a></span> - Static variable in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/IlluminaStrand.html" title="class in edu.uab.ssg.model.snp">IlluminaStrand</a></dt>
<dd>
<div class="block">The BOT strand.</div>
</dd>
</dl>
<a name="I:C">
<!--   -->
</a>
<h2 class="title">C</h2>
<dl>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/eigenstrat/GENOWriter.html#close--">close()</a></span> - Method in class edu.uab.ssg.io.eigenstrat.<a href="edu/uab/ssg/io/eigenstrat/GENOWriter.html" title="class in edu.uab.ssg.io.eigenstrat">GENOWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/fastphase/OUTWriter.html#close--">close()</a></span> - Method in class edu.uab.ssg.io.fastphase.<a href="edu/uab/ssg/io/fastphase/OUTWriter.html" title="class in edu.uab.ssg.io.fastphase">OUTWriter</a></dt>
<dd>
<div class="block">Closes the writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hyperlasso/DATWriter.html#close--">close()</a></span> - Method in class edu.uab.ssg.io.hyperlasso.<a href="edu/uab/ssg/io/hyperlasso/DATWriter.html" title="class in edu.uab.ssg.io.hyperlasso">DATWriter</a></dt>
<dd>
<div class="block">Closes the writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSWriter.html#close--">close()</a></span> - Method in class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSWriter.html" title="class in edu.uab.ssg.io.marchini_gwas">GENSWriter</a></dt>
<dd>
<div class="block">Closes the writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDWriter.html#close--">close()</a></span> - Method in class edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDWriter.html" title="class in edu.uab.ssg.io.plink">PEDWriter</a></dt>
<dd>
<div class="block">Closes the writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/LegendBuilder.html#countAllele-edu.uab.ssg.model.snp.SNP-java.lang.String-">countAllele(SNP, String)</a></span> - Method in class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/LegendBuilder.html" title="class in edu.uab.ssg.io.marchini_gwas">LegendBuilder</a></dt>
<dd>
<div class="block">Counts an allele at a SNP.</div>
</dd>
<dt><a href="edu/uab/ssg/io/limdi/CovariateTabParser.html" title="class in edu.uab.ssg.io.limdi"><span class="typeNameLink">CovariateTabParser</span></a> - Class in <a href="edu/uab/ssg/io/limdi/package-summary.html">edu.uab.ssg.io.limdi</a></dt>
<dd>
<div class="block">A parser for the tab-delimited text file, called <i>Data_jelai_25March_09.tab</i>, containing the covariate data for the Limdi warfarin analysis.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.html#CovariateTabParser--">CovariateTabParser()</a></span> - Constructor for class edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.html" title="class in edu.uab.ssg.io.limdi">CovariateTabParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/limdi/CovariateTabParser.RecordListener.html" title="interface in edu.uab.ssg.io.limdi"><span class="typeNameLink">CovariateTabParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/limdi/package-summary.html">edu.uab.ssg.io.limdi</a></dt>
<dd>
<div class="block">A listener for handling parsed sample records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi"><span class="typeNameLink">CovariateTabParser.SampleRecord</span></a> - Interface in <a href="edu/uab/ssg/io/limdi/package-summary.html">edu.uab.ssg.io.limdi</a></dt>
<dd>
<div class="block">A sample record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/LegendBuilder.html#createMinorAlleleLegend--">createMinorAlleleLegend()</a></span> - Method in class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/LegendBuilder.html" title="class in edu.uab.ssg.io.marchini_gwas">LegendBuilder</a></dt>
<dd>
<div class="block">Creates legend that recodes alleles to 0 or 1 based on the minor allele.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/LegendBuilder.html#createReferenceSampleLegend-edu.uab.ssg.model.snp.Sample-">createReferenceSampleLegend(Sample)</a></span> - Method in class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/LegendBuilder.html" title="class in edu.uab.ssg.io.marchini_gwas">LegendBuilder</a></dt>
<dd>
<div class="block">Creates legend that recodes alleles to 0 or 1 based on the haplotype of a given reference sample.</div>
</dd>
</dl>
<a name="I:D">
<!--   -->
</a>
<h2 class="title">D</h2>
<dl>
<dt><a href="edu/uab/ssg/io/hyperlasso/DATWriter.html" title="class in edu.uab.ssg.io.hyperlasso"><span class="typeNameLink">DATWriter</span></a> - Class in <a href="edu/uab/ssg/io/hyperlasso/package-summary.html">edu.uab.ssg.io.hyperlasso</a></dt>
<dd>
<div class="block">A writer for the HyperLasso DAT file format described in <a href="http://www.ebi.ac.uk/projects/BARGEN/download/HyperLasso/readme.txt">this README</a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hyperlasso/DATWriter.html#DATWriter-java.util.List-java.io.OutputStream-">DATWriter(List&lt;SNP&gt;, OutputStream)</a></span> - Constructor for class edu.uab.ssg.io.hyperlasso.<a href="edu/uab/ssg/io/hyperlasso/DATWriter.html" title="class in edu.uab.ssg.io.hyperlasso">DATWriter</a></dt>
<dd>
<div class="block">Constructs a writer.</div>
</dd>
<dt><a href="edu/uab/ssg/model/snp/DefaultSNP.html" title="class in edu.uab.ssg.model.snp"><span class="typeNameLink">DefaultSNP</span></a> - Class in <a href="edu/uab/ssg/model/snp/package-summary.html">edu.uab.ssg.model.snp</a></dt>
<dd>
<div class="block">A default SNP implementation that defines equals() and hashCode() in terms of chromosome location, not SNP name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/DefaultSNP.html#DefaultSNP-java.lang.String-java.lang.String-int-">DefaultSNP(String, String, int)</a></span> - Constructor for class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/DefaultSNP.html" title="class in edu.uab.ssg.model.snp">DefaultSNP</a></dt>
<dd>
<div class="block">Constructs a SNP.</div>
</dd>
<dt><a href="edu/uab/ssg/io/limdi/DoseTabParser.html" title="class in edu.uab.ssg.io.limdi"><span class="typeNameLink">DoseTabParser</span></a> - Class in <a href="edu/uab/ssg/io/limdi/package-summary.html">edu.uab.ssg.io.limdi</a></dt>
<dd>
<div class="block">A parser for the tab-delimited text file, called <i>GWAS_AA_feb19_09_B109.tab</i>, containing the dose data for the Limdi warfarin analysis.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/DoseTabParser.html#DoseTabParser--">DoseTabParser()</a></span> - Constructor for class edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/DoseTabParser.html" title="class in edu.uab.ssg.io.limdi">DoseTabParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/limdi/DoseTabParser.RecordListener.html" title="interface in edu.uab.ssg.io.limdi"><span class="typeNameLink">DoseTabParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/limdi/package-summary.html">edu.uab.ssg.io.limdi</a></dt>
<dd>
<div class="block">A listener for handling parsed sample records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/limdi/DoseTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi"><span class="typeNameLink">DoseTabParser.SampleRecord</span></a> - Interface in <a href="edu/uab/ssg/io/limdi/package-summary.html">edu.uab.ssg.io.limdi</a></dt>
<dd>
<div class="block">A sample record.</div>
</dd>
</dl>
<a name="I:E">
<!--   -->
</a>
<h2 class="title">E</h2>
<dl>
<dt><a href="edu/uab/ssg/io/decode/package-summary.html">edu.uab.ssg.io.decode</a> - package edu.uab.ssg.io.decode</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/eigenstrat/package-summary.html">edu.uab.ssg.io.eigenstrat</a> - package edu.uab.ssg.io.eigenstrat</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/ent/package-summary.html">edu.uab.ssg.io.ent</a> - package edu.uab.ssg.io.ent</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/fastphase/package-summary.html">edu.uab.ssg.io.fastphase</a> - package edu.uab.ssg.io.fastphase</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/haploview/package-summary.html">edu.uab.ssg.io.haploview</a> - package edu.uab.ssg.io.haploview</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/hapmap/package-summary.html">edu.uab.ssg.io.hapmap</a> - package edu.uab.ssg.io.hapmap</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/hyperlasso/package-summary.html">edu.uab.ssg.io.hyperlasso</a> - package edu.uab.ssg.io.hyperlasso</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/illumina/package-summary.html">edu.uab.ssg.io.illumina</a> - package edu.uab.ssg.io.illumina</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/limdi/package-summary.html">edu.uab.ssg.io.limdi</a> - package edu.uab.ssg.io.limdi</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/mach/package-summary.html">edu.uab.ssg.io.mach</a> - package edu.uab.ssg.io.mach</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/marchini_gwas/package-summary.html">edu.uab.ssg.io.marchini_gwas</a> - package edu.uab.ssg.io.marchini_gwas</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/ncbi/package-summary.html">edu.uab.ssg.io.ncbi</a> - package edu.uab.ssg.io.ncbi</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/omrf/package-summary.html">edu.uab.ssg.io.omrf</a> - package edu.uab.ssg.io.omrf</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/phase/package-summary.html">edu.uab.ssg.io.phase</a> - package edu.uab.ssg.io.phase</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/plink/package-summary.html">edu.uab.ssg.io.plink</a> - package edu.uab.ssg.io.plink</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/the1000genomes/package-summary.html">edu.uab.ssg.io.the1000genomes</a> - package edu.uab.ssg.io.the1000genomes</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/ucsc/package-summary.html">edu.uab.ssg.io.ucsc</a> - package edu.uab.ssg.io.ucsc</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/model/snp/package-summary.html">edu.uab.ssg.model.snp</a> - package edu.uab.ssg.model.snp</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/reports/package-summary.html">edu.uab.ssg.reports</a> - package edu.uab.ssg.reports</dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/ent/ENTWriter.html" title="class in edu.uab.ssg.io.ent"><span class="typeNameLink">ENTWriter</span></a> - Class in <a href="edu/uab/ssg/io/ent/package-summary.html">edu.uab.ssg.io.ent</a></dt>
<dd>
<div class="block">A writer for the ENT <a href="http://dna.engr.uconn.edu/~software/ent/">http://dna.engr.uconn.edu/~software/ent/</a> input file format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ent/ENTWriter.html#ENTWriter--">ENTWriter()</a></span> - Constructor for class edu.uab.ssg.io.ent.<a href="edu/uab/ssg/io/ent/ENTWriter.html" title="class in edu.uab.ssg.io.ent">ENTWriter</a></dt>
<dd>
<div class="block">Constructs the writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/DefaultSNP.html#equals-java.lang.Object-">equals(Object)</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/DefaultSNP.html" title="class in edu.uab.ssg.model.snp">DefaultSNP</a></dt>
<dd>
<div class="block">Returns true if the user-supplied object is a SNP at the same chromosome
 position.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/AlleleCounter.html#existsAllele-java.lang.String-">existsAllele(String)</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/AlleleCounter.html" title="class in edu.uab.ssg.model.snp">AlleleCounter</a></dt>
<dd>
<div class="block">Returns true if the given allele has been counted at least once.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.SNPRecord.html#existsAlleleA-java.lang.String-">existsAlleleA(String)</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.hapmap">HaplotypeFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns true if the A allele for the given sample exists.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.SNPRecord.html#existsAlleleB-java.lang.String-">existsAlleleB(String)</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.hapmap">HaplotypeFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns true if the B allele for the given sample exists.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GTypeFileParser.SNPRecord.html#existsGenotype-java.lang.String-">existsGenotype(String)</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GTypeFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">GTypeFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns true if this record contains a genotype (at this SNP) for the given sample.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/Sample.html#existsGenotype-edu.uab.ssg.model.snp.SNP-">existsGenotype(SNP)</a></span> - Method in interface edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/Sample.html" title="interface in edu.uab.ssg.model.snp">Sample</a></dt>
<dd>
<div class="block">Returns true if genotype for the given SNP was assessed in this sample.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/AlleleCounter.html#existsMinorAllele--">existsMinorAllele()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/AlleleCounter.html" title="class in edu.uab.ssg.model.snp">AlleleCounter</a></dt>
<dd>
<div class="block">Returns true if the minor allele exists.</div>
</dd>
</dl>
<a name="I:F">
<!--   -->
</a>
<h2 class="title">F</h2>
<dl>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/Sex.html#FEMALE">FEMALE</a></span> - Static variable in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/Sex.html" title="class in edu.uab.ssg.model.snp">Sex</a></dt>
<dd>
<div class="block">The female sex.</div>
</dd>
<dt><a href="edu/uab/ssg/reports/FlipCalls.html" title="class in edu.uab.ssg.reports"><span class="typeNameLink">FlipCalls</span></a> - Class in <a href="edu/uab/ssg/reports/package-summary.html">edu.uab.ssg.reports</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/reports/FlipCalls.html#FlipCalls--">FlipCalls()</a></span> - Constructor for class edu.uab.ssg.reports.<a href="edu/uab/ssg/reports/FlipCalls.html" title="class in edu.uab.ssg.reports">FlipCalls</a></dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/decode/FullScoreFileParser.html" title="class in edu.uab.ssg.io.decode"><span class="typeNameLink">FullScoreFileParser</span></a> - Class in <a href="edu/uab/ssg/io/decode/package-summary.html">edu.uab.ssg.io.decode</a></dt>
<dd>
<div class="block">A parser for the <i>snps_fullscore_20080328.tab</i> file compiled by Gisli 
 Masson at deCODE and circulated to the UAB popgen team on March 28, 2008.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.html#FullScoreFileParser--">FullScoreFileParser()</a></span> - Constructor for class edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.html" title="class in edu.uab.ssg.io.decode">FullScoreFileParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/decode/FullScoreFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.decode"><span class="typeNameLink">FullScoreFileParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/decode/package-summary.html">edu.uab.ssg.io.decode</a></dt>
<dd>
<div class="block">A listener for handling parsed SNP records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode"><span class="typeNameLink">FullScoreFileParser.SNPRecord</span></a> - Interface in <a href="edu/uab/ssg/io/decode/package-summary.html">edu.uab.ssg.io.decode</a></dt>
<dd>
<div class="block">A SNP record.</div>
</dd>
</dl>
<a name="I:G">
<!--   -->
</a>
<h2 class="title">G</h2>
<dl>
<dt><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.html" title="class in edu.uab.ssg.io.ncbi"><span class="typeNameLink">GeneInfoParser</span></a> - Class in <a href="edu/uab/ssg/io/ncbi/package-summary.html">edu.uab.ssg.io.ncbi</a></dt>
<dd>
<div class="block">A parser for the Entrez Gene gene_info file located at <a href="./edu/uab/ssg/io/ncbi/ftp://ftp.ncbi.nih.gov/gene/DATA/gene_info.gz">ftp://ftp.ncbi.nih.gov/gene/DATA/gene_info.gz</a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.html#GeneInfoParser--">GeneInfoParser()</a></span> - Constructor for class edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.html" title="class in edu.uab.ssg.io.ncbi">GeneInfoParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi"><span class="typeNameLink">GeneInfoParser.Record</span></a> - Interface in <a href="edu/uab/ssg/io/ncbi/package-summary.html">edu.uab.ssg.io.ncbi</a></dt>
<dd>
<div class="block">A gene info record.</div>
</dd>
<dt><a href="edu/uab/ssg/reports/GeneReport.html" title="class in edu.uab.ssg.reports"><span class="typeNameLink">GeneReport</span></a> - Class in <a href="edu/uab/ssg/reports/package-summary.html">edu.uab.ssg.reports</a></dt>
<dd>
<div class="block">This gene report parses NCBI-formatted <tt>gene_info</tt> and <tt>seq_gene.md</tt> files and outputs a tab-delimited text table containing annotation records for each user-supplied gene of interest.</div>
</dd>
<dt><a href="edu/uab/ssg/io/mach/GENOParser.html" title="class in edu.uab.ssg.io.mach"><span class="typeNameLink">GENOParser</span></a> - Class in <a href="edu/uab/ssg/io/mach/package-summary.html">edu.uab.ssg.io.mach</a></dt>
<dd>
<div class="block">A parser for the GENO output file format from <a href="http://www.sph.umich.edu/csg/abecasis/MaCH/">MACH</a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/mach/GENOParser.html#GENOParser--">GENOParser()</a></span> - Constructor for class edu.uab.ssg.io.mach.<a href="edu/uab/ssg/io/mach/GENOParser.html" title="class in edu.uab.ssg.io.mach">GENOParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/mach/GENOParser.IndividualRecord.html" title="interface in edu.uab.ssg.io.mach"><span class="typeNameLink">GENOParser.IndividualRecord</span></a> - Interface in <a href="edu/uab/ssg/io/mach/package-summary.html">edu.uab.ssg.io.mach</a></dt>
<dd>
<div class="block">An individual record.</div>
</dd>
<dt><a href="edu/uab/ssg/io/mach/GENOParser.RecordListener.html" title="interface in edu.uab.ssg.io.mach"><span class="typeNameLink">GENOParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/mach/package-summary.html">edu.uab.ssg.io.mach</a></dt>
<dd>
<div class="block">A listener for handling parsed individual records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/decode/GenotypeFileParser.html" title="class in edu.uab.ssg.io.decode"><span class="typeNameLink">GenotypeFileParser</span></a> - Class in <a href="edu/uab/ssg/io/decode/package-summary.html">edu.uab.ssg.io.decode</a></dt>
<dd>
<div class="block">A parser for the genotype file format used by deCODE to report genotyping
 results from a custom Illumina SNP array.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GenotypeFileParser.html#GenotypeFileParser--">GenotypeFileParser()</a></span> - Constructor for class edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GenotypeFileParser.html" title="class in edu.uab.ssg.io.decode">GenotypeFileParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.html" title="class in edu.uab.ssg.io.hapmap"><span class="typeNameLink">GenotypeFileParser</span></a> - Class in <a href="edu/uab/ssg/io/hapmap/package-summary.html">edu.uab.ssg.io.hapmap</a></dt>
<dd>
<div class="block">A parser for the genotype file format for the HapMap phases I+II+III release #28 (NCBI build 36, dbSNP b126) data located at <a href="http://hapmap.ncbi.nlm.nih.gov/downloads/genotypes/2010-08_phaseII+III/">http://hapmap.ncbi.nlm.nih.gov/downloads/genotypes/2010-08_phaseII+III/</a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.html#GenotypeFileParser--">GenotypeFileParser()</a></span> - Constructor for class edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.html" title="class in edu.uab.ssg.io.hapmap">GenotypeFileParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/illumina/GenotypeFileParser.html" title="class in edu.uab.ssg.io.illumina"><span class="typeNameLink">GenotypeFileParser</span></a> - Class in <a href="edu/uab/ssg/io/illumina/package-summary.html">edu.uab.ssg.io.illumina</a></dt>
<dd>
<div class="block">A parser that can flexibly handle the various formats that Illumina genotyping reports can have.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/GenotypeFileParser.html#GenotypeFileParser--">GenotypeFileParser()</a></span> - Constructor for class edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/GenotypeFileParser.html" title="class in edu.uab.ssg.io.illumina">GenotypeFileParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/limdi/GenotypeFileParser.html" title="class in edu.uab.ssg.io.limdi"><span class="typeNameLink">GenotypeFileParser</span></a> - Class in <a href="edu/uab/ssg/io/limdi/package-summary.html">edu.uab.ssg.io.limdi</a></dt>
<dd>
<div class="block">A parser for the genotype file format used by the Genome Sciences Genotyping and Resequencing Center (GS GRC) to report genotyping results from an Illumina Infinium 1M Duo V3 SNP array.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/GenotypeFileParser.html#GenotypeFileParser--">GenotypeFileParser()</a></span> - Constructor for class edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/GenotypeFileParser.html" title="class in edu.uab.ssg.io.limdi">GenotypeFileParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/decode/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.decode"><span class="typeNameLink">GenotypeFileParser.GenotypeRecord</span></a> - Interface in <a href="edu/uab/ssg/io/decode/package-summary.html">edu.uab.ssg.io.decode</a></dt>
<dd>
<div class="block">A genotype record.</div>
</dd>
<dt><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.hapmap"><span class="typeNameLink">GenotypeFileParser.GenotypeRecord</span></a> - Interface in <a href="edu/uab/ssg/io/hapmap/package-summary.html">edu.uab.ssg.io.hapmap</a></dt>
<dd>
<div class="block">A genotype record.</div>
</dd>
<dt><a href="edu/uab/ssg/io/illumina/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.illumina"><span class="typeNameLink">GenotypeFileParser.GenotypeRecord</span></a> - Interface in <a href="edu/uab/ssg/io/illumina/package-summary.html">edu.uab.ssg.io.illumina</a></dt>
<dd>
<div class="block">A genotype record.</div>
</dd>
<dt><a href="edu/uab/ssg/io/limdi/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.limdi"><span class="typeNameLink">GenotypeFileParser.GenotypeRecord</span></a> - Interface in <a href="edu/uab/ssg/io/limdi/package-summary.html">edu.uab.ssg.io.limdi</a></dt>
<dd>
<div class="block">A genotype record.</div>
</dd>
<dt><a href="edu/uab/ssg/io/decode/GenotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.decode"><span class="typeNameLink">GenotypeFileParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/decode/package-summary.html">edu.uab.ssg.io.decode</a></dt>
<dd>
<div class="block">A listener for handling parsed genotype records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.hapmap"><span class="typeNameLink">GenotypeFileParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/hapmap/package-summary.html">edu.uab.ssg.io.hapmap</a></dt>
<dd>
<div class="block">A listener for handling parsed genotype records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/illumina/GenotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.illumina"><span class="typeNameLink">GenotypeFileParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/illumina/package-summary.html">edu.uab.ssg.io.illumina</a></dt>
<dd>
<div class="block">A listener for handling parsed genotype records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/limdi/GenotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.limdi"><span class="typeNameLink">GenotypeFileParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/limdi/package-summary.html">edu.uab.ssg.io.limdi</a></dt>
<dd>
<div class="block">A listener for handling parsed genotype records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/eigenstrat/GENOWriter.html" title="class in edu.uab.ssg.io.eigenstrat"><span class="typeNameLink">GENOWriter</span></a> - Class in <a href="edu/uab/ssg/io/eigenstrat/package-summary.html">edu.uab.ssg.io.eigenstrat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/eigenstrat/GENOWriter.html#GENOWriter-java.io.OutputStream-">GENOWriter(OutputStream)</a></span> - Constructor for class edu.uab.ssg.io.eigenstrat.<a href="edu/uab/ssg/io/eigenstrat/GENOWriter.html" title="class in edu.uab.ssg.io.eigenstrat">GENOWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.html" title="class in edu.uab.ssg.io.marchini_gwas"><span class="typeNameLink">GENSParser</span></a> - Class in <a href="edu/uab/ssg/io/marchini_gwas/package-summary.html">edu.uab.ssg.io.marchini_gwas</a></dt>
<dd>
<div class="block">A parser for the genotype file format described at <a href="http://www.stats.ox.ac.uk/~marchini/software/gwas/file_format.html">http://www.stats.ox.ac.uk/~marchini/software/gwas/file_format.html</a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.html#GENSParser--">GENSParser()</a></span> - Constructor for class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.html" title="class in edu.uab.ssg.io.marchini_gwas">GENSParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.Genotype.html" title="interface in edu.uab.ssg.io.marchini_gwas"><span class="typeNameLink">GENSParser.Genotype</span></a> - Interface in <a href="edu/uab/ssg/io/marchini_gwas/package-summary.html">edu.uab.ssg.io.marchini_gwas</a></dt>
<dd>
<div class="block">A sample genotype.</div>
</dd>
<dt><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeProbabilities.html" title="interface in edu.uab.ssg.io.marchini_gwas"><span class="typeNameLink">GENSParser.GenotypeProbabilities</span></a> - Interface in <a href="edu/uab/ssg/io/marchini_gwas/package-summary.html">edu.uab.ssg.io.marchini_gwas</a></dt>
<dd>
<div class="block">A set of genotype probabilities for a sample.</div>
</dd>
<dt><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.marchini_gwas"><span class="typeNameLink">GENSParser.GenotypeRecord</span></a> - Interface in <a href="edu/uab/ssg/io/marchini_gwas/package-summary.html">edu.uab.ssg.io.marchini_gwas</a></dt>
<dd>
<div class="block">A record of sample genotypes at a particular SNP.</div>
</dd>
<dt><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.RecordListener.html" title="interface in edu.uab.ssg.io.marchini_gwas"><span class="typeNameLink">GENSParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/marchini_gwas/package-summary.html">edu.uab.ssg.io.marchini_gwas</a></dt>
<dd>
<div class="block">A listener for handling parsed genotype records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/marchini_gwas/GENSWriter.html" title="class in edu.uab.ssg.io.marchini_gwas"><span class="typeNameLink">GENSWriter</span></a> - Class in <a href="edu/uab/ssg/io/marchini_gwas/package-summary.html">edu.uab.ssg.io.marchini_gwas</a></dt>
<dd>
<div class="block">A writer for the genotype file format described at <a href="http://www.stats.ox.ac.uk/~marchini/software/gwas/file_format.html">http://www.stats.ox.ac.uk/~marchini/software/gwas/file_format.html</a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSWriter.html#GENSWriter-java.util.List-java.io.OutputStream-">GENSWriter(List&lt;Sample&gt;, OutputStream)</a></span> - Constructor for class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSWriter.html" title="class in edu.uab.ssg.io.marchini_gwas">GENSWriter</a></dt>
<dd>
<div class="block">Constructs the writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getAge--">getAge()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html#getAge--">getAge()</a></span> - Method in interface edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.omrf">TabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getalcohol--">getalcohol()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html#getAllele-java.lang.String-">getAllele(String)</a></span> - Method in interface edu.uab.ssg.io.the1000genomes.<a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html" title="interface in edu.uab.ssg.io.the1000genomes">VCFParser.VariantRecord</a></dt>
<dd>
<div class="block">Returns the reference or alternate allele, corresponding to the given GT code.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/Legend.html#getAllele0-edu.uab.ssg.model.snp.SNP-">getAllele0(SNP)</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/Legend.html" title="interface in edu.uab.ssg.io.marchini_gwas">Legend</a></dt>
<dd>
<div class="block">Returns the allele coded as allele 0 at this SNP.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/LegendFile.html#getAllele0-edu.uab.ssg.model.snp.SNP-">getAllele0(SNP)</a></span> - Method in class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/LegendFile.html" title="class in edu.uab.ssg.io.marchini_gwas">LegendFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html#getAllele1--">getAllele1()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.hapmap">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the first allele observed on the strand for this sample or null if the data are missing.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.Genotype.html#getAllele1--">getAllele1()</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.Genotype.html" title="interface in edu.uab.ssg.io.marchini_gwas">GENSParser.Genotype</a></dt>
<dd>
<div class="block">Returns the first allele.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/Legend.html#getAllele1-edu.uab.ssg.model.snp.SNP-">getAllele1(SNP)</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/Legend.html" title="interface in edu.uab.ssg.io.marchini_gwas">Legend</a></dt>
<dd>
<div class="block">Returns the allele coded as allele 1 at this SNP.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/LegendFile.html#getAllele1-edu.uab.ssg.model.snp.SNP-">getAllele1(SNP)</a></span> - Method in class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/LegendFile.html" title="class in edu.uab.ssg.io.marchini_gwas">LegendFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html#getAllele1-int-">getAllele1(int)</a></span> - Method in interface edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html" title="interface in edu.uab.ssg.io.plink">PEDParser.SampleRecord</a></dt>
<dd>
<div class="block">Returns allele1 at the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html#getAllele1-java.lang.String-">getAllele1(String)</a></span> - Method in interface edu.uab.ssg.io.the1000genomes.<a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html" title="interface in edu.uab.ssg.io.the1000genomes">VCFParser.VariantRecord</a></dt>
<dd>
<div class="block">Returns allele 1 for the given sample for this variant.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/Sample.Genotype.html#getAllele1--">getAllele1()</a></span> - Method in interface edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/Sample.Genotype.html" title="interface in edu.uab.ssg.model.snp">Sample.Genotype</a></dt>
<dd>
<div class="block">Returns the first allele of this genotype.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/fastphase/OUTParser.SampleRecord.html#getAllele1At-int-">getAllele1At(int)</a></span> - Method in interface edu.uab.ssg.io.fastphase.<a href="edu/uab/ssg/io/fastphase/OUTParser.SampleRecord.html" title="interface in edu.uab.ssg.io.fastphase">OUTParser.SampleRecord</a></dt>
<dd>
<div class="block">Returns the allele at the given SNP index on the first haplotype.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GenotypeFileParser.GenotypeRecord.html#getAllele1Forward--">getAllele1Forward()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.decode">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the first allele from the forward strand.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/GenotypeFileParser.GenotypeRecord.html#getAllele1Forward--">getAllele1Forward()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.limdi">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the first allele from the forward strand or null if the data are missing.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GenotypeFileParser.GenotypeRecord.html#getAllele1Top--">getAllele1Top()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.decode">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the first allele from the TOP strand.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html#getAllele2--">getAllele2()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.hapmap">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the second allele observed on the strand for this sample or null if the data are missing.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.Genotype.html#getAllele2--">getAllele2()</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.Genotype.html" title="interface in edu.uab.ssg.io.marchini_gwas">GENSParser.Genotype</a></dt>
<dd>
<div class="block">Returns the second allele.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html#getAllele2-int-">getAllele2(int)</a></span> - Method in interface edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html" title="interface in edu.uab.ssg.io.plink">PEDParser.SampleRecord</a></dt>
<dd>
<div class="block">Returns allele2 at the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html#getAllele2-java.lang.String-">getAllele2(String)</a></span> - Method in interface edu.uab.ssg.io.the1000genomes.<a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html" title="interface in edu.uab.ssg.io.the1000genomes">VCFParser.VariantRecord</a></dt>
<dd>
<div class="block">Returns allele 2 for the given sample for this variant.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/Sample.Genotype.html#getAllele2--">getAllele2()</a></span> - Method in interface edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/Sample.Genotype.html" title="interface in edu.uab.ssg.model.snp">Sample.Genotype</a></dt>
<dd>
<div class="block">Returns the second allele of this genotype.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/fastphase/OUTParser.SampleRecord.html#getAllele2At-int-">getAllele2At(int)</a></span> - Method in interface edu.uab.ssg.io.fastphase.<a href="edu/uab/ssg/io/fastphase/OUTParser.SampleRecord.html" title="interface in edu.uab.ssg.io.fastphase">OUTParser.SampleRecord</a></dt>
<dd>
<div class="block">Returns the allele at the given SNP index on the second haplotype.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GenotypeFileParser.GenotypeRecord.html#getAllele2Forward--">getAllele2Forward()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.decode">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the second allele from the forward strand.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/GenotypeFileParser.GenotypeRecord.html#getAllele2Forward--">getAllele2Forward()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.limdi">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the second allele from the forward strand or null if the data are missing.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GenotypeFileParser.GenotypeRecord.html#getAllele2Top--">getAllele2Top()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.decode">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the second allele from the TOP strand.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.SNPRecord.html#getAlleleA-java.lang.String-">getAlleleA(String)</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.hapmap">HaplotypeFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the A allele for the given sample.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeRecord.html#getAlleleA--">getAlleleA()</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.marchini_gwas">GENSParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns allele A.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.SNPRecord.html#getAlleleB-java.lang.String-">getAlleleB(String)</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.hapmap">HaplotypeFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the B allele for the given sample.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeRecord.html#getAlleleB--">getAlleleB()</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.marchini_gwas">GENSParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns allele B.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html#getAlleles--">getAlleles()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.hapmap">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the observed alleles for the strand.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/AlleleCounter.html#getAlleles--">getAlleles()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/AlleleCounter.html" title="class in edu.uab.ssg.model.snp">AlleleCounter</a></dt>
<dd>
<div class="block">Returns the set of counted allele values.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/LegendBuilder.html#getAllelesForBadSNP-edu.uab.ssg.model.snp.SNP-">getAllelesForBadSNP(SNP)</a></span> - Method in class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/LegendBuilder.html" title="class in edu.uab.ssg.io.marchini_gwas">LegendBuilder</a></dt>
<dd>
<div class="block">For a SNP that is not monomorphic or biallelic, return the set of observed alleles for troubleshooting purposes.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html#getAlternateAllele--">getAlternateAllele()</a></span> - Method in interface edu.uab.ssg.io.the1000genomes.<a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html" title="interface in edu.uab.ssg.io.the1000genomes">VCFParser.VariantRecord</a></dt>
<dd>
<div class="block">Returns the alternate, non-reference allele.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getamiod--">getamiod()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getAssayLSID--">getAssayLSID()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the LSID for HapMap assay used for genotyping.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html#getAssemblyVersion--">getAssemblyVersion()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.hapmap">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the genome assembly version.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getAvg1Maint--">getAvg1Maint()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/DoseTabParser.SampleRecord.html#getAvgDose--">getAvgDose()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/DoseTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">DoseTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getAvgMaint--">getAvgMaint()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getAvgOfK--">getAvgOfK()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/LegendBuilder.html#getBadSNPs--">getBadSNPs()</a></span> - Method in class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/LegendBuilder.html" title="class in edu.uab.ssg.io.marchini_gwas">LegendBuilder</a></dt>
<dd>
<div class="block">Returns a list of SNPs that are not monomorphic or biallelic.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getBMI--">getBMI()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getBuild--">getBuild()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the reference sequence build version.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getC2C9--">getC2C9()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getCenter--">getCenter()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the genotyping center that submitted underlying genotypes.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html#getChr--">getChr()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">FullScoreFileParser.SNPRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html#getchr--">getchr()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">FullScoreFileParser.SNPRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GTypeFileParser.SNPRecord.html#getChr--">getChr()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GTypeFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">GTypeFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the chromosome name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/SNPFileParser.SNPRecord.html#getChr--">getChr()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">SNPFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the chromosome name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getChrom--">getChrom()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the chromosome the SNP maps to on current reference sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html#getChrom--">getChrom()</a></span> - Method in interface edu.uab.ssg.io.ucsc.<a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html" title="interface in edu.uab.ssg.io.ucsc">SNPAnnotationParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the chromosome.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html#getChromEnd--">getChromEnd()</a></span> - Method in interface edu.uab.ssg.io.ucsc.<a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html" title="interface in edu.uab.ssg.io.ucsc">SNPAnnotationParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the end position on the chromosome.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html#getChromosome--">getChromosome()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.hapmap">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the chromosome.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html#getChromosome--">getChromosome()</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.illumina">SNPFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the chromosome name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeRecord.html#getChromosome--">getChromosome()</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.marchini_gwas">GENSParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the chromosome.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html#getChromosome--">getChromosome()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">GeneInfoParser.Record</a></dt>
<dd>
<div class="block">Returns the chromosome on which this gene is located.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html#getChromosome--">getChromosome()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.Record</a></dt>
<dd>
<div class="block">Returns the chromosome name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html#getChromosome--">getChromosome()</a></span> - Method in interface edu.uab.ssg.io.the1000genomes.<a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html" title="interface in edu.uab.ssg.io.the1000genomes">VCFParser.VariantRecord</a></dt>
<dd>
<div class="block">Returns the chromosome.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/DefaultSNP.html#getChromosome--">getChromosome()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/DefaultSNP.html" title="class in edu.uab.ssg.model.snp">DefaultSNP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/SNP.html#getChromosome--">getChromosome()</a></span> - Method in interface edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/SNP.html" title="interface in edu.uab.ssg.model.snp">SNP</a></dt>
<dd>
<div class="block">Returns the chromosome where the SNP is located.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html#getChromStart--">getChromStart()</a></span> - Method in interface edu.uab.ssg.io.ucsc.<a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html" title="interface in edu.uab.ssg.io.ucsc">SNPAnnotationParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the start position on the chromosome.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html#getChrOrient--">getChrOrient()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.Record</a></dt>
<dd>
<div class="block">Returns the orientation of this feature on the chromosome for this asembly.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html#getChrStart--">getChrStart()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.Record</a></dt>
<dd>
<div class="block">Returns the start position of this feature on the chromosome for this assembly.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html#getChrStop--">getChrStop()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.Record</a></dt>
<dd>
<div class="block">Returns the stop position of this feature on the chromosome for this assembly.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getCKD--">getCKD()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getCKD1--">getCKD1()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/GenotypeFileParser.GenotypeRecord.html#getColumnNames--">getColumnNames()</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.illumina">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the list of column names available for this record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html#getContig--">getContig()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.Record</a></dt>
<dd>
<div class="block">Returns the contig name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html#getCoordinate--">getCoordinate()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">FullScoreFileParser.SNPRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html#getCtgOrient--">getCtgOrient()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.Record</a></dt>
<dd>
<div class="block">Returns the orientation of this feature on the contig.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html#getCtgStart--">getCtgStart()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.Record</a></dt>
<dd>
<div class="block">Returns the start position of this feature on the contig.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html#getCtgStop--">getCtgStop()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.Record</a></dt>
<dd>
<div class="block">Returns the stop position of this feature on the contig.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/SNPFileParser.SNPRecord.html#getCustomerStrand--">getCustomerStrand()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">SNPFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the strand of the allele reported by the customer in 
 Illumina TOP/BOT strand notation.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html#getCustomerStrand--">getCustomerStrand()</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.illumina">SNPFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the strand of the allele reported by the customer in 
 Illumina TOP/BOT strand notation.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getCYP2C9--">getCYP2C9()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html#getDateLastVacc--">getDateLastVacc()</a></span> - Method in interface edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.omrf">TabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html#getDateReceived--">getDateReceived()</a></span> - Method in interface edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.omrf">TabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html#getdbXrefs--">getdbXrefs()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">GeneInfoParser.Record</a></dt>
<dd>
<div class="block">Returns a pipe-delimited set of identifiers, of the form <code>database:value</code>, in other databases for this gene.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html#getDescription--">getDescription()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">GeneInfoParser.Record</a></dt>
<dd>
<div class="block">Returns a description for this gene.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html#getEFEndTiter--">getEFEndTiter()</a></span> - Method in interface edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.omrf">TabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html#getEvidenceCode--">getEvidenceCode()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.Record</a></dt>
<dd>
<div class="block">Returns the evidence code.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/mach/GENOParser.IndividualRecord.html#getFamilyID--">getFamilyID()</a></span> - Method in interface edu.uab.ssg.io.mach.<a href="edu/uab/ssg/io/mach/GENOParser.IndividualRecord.html" title="interface in edu.uab.ssg.io.mach">GENOParser.IndividualRecord</a></dt>
<dd>
<div class="block">Returns the family ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html#getFeatureID--">getFeatureID()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.Record</a></dt>
<dd>
<div class="block">Returns the ID of this feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html#getFeatureName--">getFeatureName()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.Record</a></dt>
<dd>
<div class="block">Returns the name of this feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html#getFeatureType--">getFeatureType()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.Record</a></dt>
<dd>
<div class="block">Returns the type of this feature, usually GENE, PSEUDO, RNA, UTR, or CDS.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html#getFID--">getFID()</a></span> - Method in interface edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html" title="interface in edu.uab.ssg.io.plink">PEDParser.SampleRecord</a></dt>
<dd>
<div class="block">Returns the family ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/AlleleCounter.html#getFrequency-java.lang.String-">getFrequency(String)</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/AlleleCounter.html" title="class in edu.uab.ssg.model.snp">AlleleCounter</a></dt>
<dd>
<div class="block">Returns the frequency, or count, of a particular allele within
 the study population.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html#getFullNameFromNomenclatureAuthority--">getFullNameFromNomenclatureAuthority()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">GeneInfoParser.Record</a></dt>
<dd>
<div class="block">Returns the full name from nomenclature authority, if available.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html#getFunc--">getFunc()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">FullScoreFileParser.SNPRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html#getFunc--">getFunc()</a></span> - Method in interface edu.uab.ssg.io.ucsc.<a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html" title="interface in edu.uab.ssg.io.ucsc">SNPAnnotationParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the functional annotation (coding-synon, intron, ..., splice-3, splice-5).</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GenotypeFileParser.GenotypeRecord.html#getGCScore--">getGCScore()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.decode">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the GenCall (GC) score for this SNP and sample.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getGender--">getGender()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/DoseTabParser.SampleRecord.html#getGender--">getGender()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/DoseTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">DoseTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html#getGeneID--">getGeneID()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">GeneInfoParser.Record</a></dt>
<dd>
<div class="block">Returns the unique identifier for this gene.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GTypeFileParser.SNPRecord.html#getGenotype-java.lang.String-">getGenotype(String)</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GTypeFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">GTypeFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the genotype at this SNP for the given sample.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeProbabilities.html#getGenotype-double-">getGenotype(double)</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeProbabilities.html" title="interface in edu.uab.ssg.io.marchini_gwas">GENSParser.GenotypeProbabilities</a></dt>
<dd>
<div class="block">Returns the genotype with probability greater than or equal to the user-specified threshold or null if none of the genotypes meet the probability threshold.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/Sample.html#getGenotype-edu.uab.ssg.model.snp.SNP-">getGenotype(SNP)</a></span> - Method in interface edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/Sample.html" title="interface in edu.uab.ssg.model.snp">Sample</a></dt>
<dd>
<div class="block">Returns the genotype of the sample for the given SNP.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeRecord.html#getGenotypeProbabilities--">getGenotypeProbabilities()</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.marchini_gwas">GENSParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the list of genotype probabilities corresponding to the sequence of samples in the sample file.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/mach/GENOParser.IndividualRecord.html#getGenotypes--">getGenotypes()</a></span> - Method in interface edu.uab.ssg.io.mach.<a href="edu/uab/ssg/io/mach/GENOParser.IndividualRecord.html" title="interface in edu.uab.ssg.io.mach">GENOParser.IndividualRecord</a></dt>
<dd>
<div class="block">Returns the genotypes as a list of strings.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html#getGenTrainScore--">getGenTrainScore()</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.illumina">SNPFileParser.SNPRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html#getGroupLabel--">getGroupLabel()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.Record</a></dt>
<dd>
<div class="block">Returns the group label, usually the name and version of the assembly.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getHeight--">getHeight()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html#getHugoSymbol--">getHugoSymbol()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">FullScoreFileParser.SNPRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html#getID--">getID()</a></span> - Method in interface edu.uab.ssg.io.the1000genomes.<a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html" title="interface in edu.uab.ssg.io.the1000genomes">VCFParser.VariantRecord</a></dt>
<dd>
<div class="block">Returns the id for this variant.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getIDNum--">getIDNum()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/DoseTabParser.SampleRecord.html#getIDNum--">getIDNum()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/DoseTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">DoseTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html#getIID--">getIID()</a></span> - Method in interface edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html" title="interface in edu.uab.ssg.io.plink">PEDParser.SampleRecord</a></dt>
<dd>
<div class="block">Returns the individual ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/SNPFileParser.SNPRecord.html#getILMNStrand--">getILMNStrand()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">SNPFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the Illumina TOP/BOT strand.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html#getILMNStrand--">getILMNStrand()</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.illumina">SNPFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the Illumina TOP/BOT strand.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/mach/GENOParser.IndividualRecord.html#getIndividualID--">getIndividualID()</a></span> - Method in interface edu.uab.ssg.io.mach.<a href="edu/uab/ssg/io/mach/GENOParser.IndividualRecord.html" title="interface in edu.uab.ssg.io.mach">GENOParser.IndividualRecord</a></dt>
<dd>
<div class="block">Returns the individual ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/SampleBuilder.html#getInstance--">getInstance()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/SampleBuilder.html" title="class in edu.uab.ssg.model.snp">SampleBuilder</a></dt>
<dd>
<div class="block">Returns the sample and disables the builder.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html#getISelectable--">getISelectable()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">FullScoreFileParser.SNPRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html#getLFEndTiter--">getLFEndTiter()</a></span> - Method in interface edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.omrf">TabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html#getLocType--">getLocType()</a></span> - Method in interface edu.uab.ssg.io.ucsc.<a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html" title="interface in edu.uab.ssg.io.ucsc">SNPAnnotationParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the "loc" (maybe location or locus) type (range, exact, between, ..., rangeSubstitution, rangeDeletion).</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html#getLocusTag--">getLocusTag()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">GeneInfoParser.Record</a></dt>
<dd>
<div class="block">Returns the LocusTag value.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getlogdose--">getlogdose()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html#getMapLocation--">getMapLocation()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">GeneInfoParser.Record</a></dt>
<dd>
<div class="block">Returns the map location for this gene.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html#getMaternalID--">getMaternalID()</a></span> - Method in interface edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html" title="interface in edu.uab.ssg.io.plink">PEDParser.SampleRecord</a></dt>
<dd>
<div class="block">Returns the maternal ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/AlleleCounter.html#getMinorAllele--">getMinorAllele()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/AlleleCounter.html" title="class in edu.uab.ssg.model.snp">AlleleCounter</a></dt>
<dd>
<div class="block">Returns the minor allele.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html#getModificationDate--">getModificationDate()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">GeneInfoParser.Record</a></dt>
<dd>
<div class="block">Returns the last date, in YYYYMMDD format, this record was updated.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html#getMolType--">getMolType()</a></span> - Method in interface edu.uab.ssg.io.ucsc.<a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html" title="interface in edu.uab.ssg.io.ucsc">SNPAnnotationParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the molecule type (unknown, genomic, or cDNA).</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GTypeFileParser.SNPRecord.html#getName--">getName()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GTypeFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">GTypeFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the SNP name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/SNPFileParser.SNPRecord.html#getName--">getName()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">SNPFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the SNP name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.SNPRecord.html#getName--">getName()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.hapmap">HaplotypeFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the SNP name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html#getName--">getName()</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.illumina">SNPFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the SNP name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeRecord.html#getName--">getName()</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.marchini_gwas">GENSParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the SNP name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html#getName--">getName()</a></span> - Method in interface edu.uab.ssg.io.ucsc.<a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html" title="interface in edu.uab.ssg.io.ucsc">SNPAnnotationParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the rsid.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/DefaultSNP.html#getName--">getName()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/DefaultSNP.html" title="class in edu.uab.ssg.model.snp">DefaultSNP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/IlluminaStrand.html#getName--">getName()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/IlluminaStrand.html" title="class in edu.uab.ssg.model.snp">IlluminaStrand</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/Sample.html#getName--">getName()</a></span> - Method in interface edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/Sample.html" title="interface in edu.uab.ssg.model.snp">Sample</a></dt>
<dd>
<div class="block">Returns the name of the sample.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/SNP.html#getName--">getName()</a></span> - Method in interface edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/SNP.html" title="interface in edu.uab.ssg.model.snp">SNP</a></dt>
<dd>
<div class="block">Returns the name of the SNP.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/Strand.html#getName--">getName()</a></span> - Method in interface edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/Strand.html" title="interface in edu.uab.ssg.model.snp">Strand</a></dt>
<dd>
<div class="block">Returns the name of the strand.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html#getNeutralizationGroup--">getNeutralizationGroup()</a></span> - Method in interface edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.omrf">TabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html#getNomenclatureStatus--">getNomenclatureStatus()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">GeneInfoParser.Record</a></dt>
<dd>
<div class="block">Returns the status (O for official, I for interim) of the name from the nomenclature authority, if available.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html#getNormID--">getNormID()</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.illumina">SNPFileParser.SNPRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html#getNumberOfAvailableGenotypeCalls--">getNumberOfAvailableGenotypeCalls()</a></span> - Method in interface edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html" title="interface in edu.uab.ssg.io.plink">PEDParser.SampleRecord</a></dt>
<dd>
<div class="block">Returns the number of genotype calls available for this sample.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/AlleleCounter.html#getNumberOfCountedValues--">getNumberOfCountedValues()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/AlleleCounter.html" title="class in edu.uab.ssg.model.snp">AlleleCounter</a></dt>
<dd>
<div class="block">Returns the number of counted, non-missing allele values.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/AlleleCounter.html#getNumberOfMissingValues--">getNumberOfMissingValues()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/AlleleCounter.html" title="class in edu.uab.ssg.model.snp">AlleleCounter</a></dt>
<dd>
<div class="block">Returns the number of missing allele values.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/fastphase/OUTParser.SampleRecord.html#getNumberOfSNPs--">getNumberOfSNPs()</a></span> - Method in interface edu.uab.ssg.io.fastphase.<a href="edu/uab/ssg/io/fastphase/OUTParser.SampleRecord.html" title="interface in edu.uab.ssg.io.fastphase">OUTParser.SampleRecord</a></dt>
<dd>
<div class="block">Returns the number of SNPs in each haplotype.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html#getNumberOfVacc--">getNumberOfVacc()</a></span> - Method in interface edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.omrf">TabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html#getObserved--">getObserved()</a></span> - Method in interface edu.uab.ssg.io.ucsc.<a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html" title="interface in edu.uab.ssg.io.ucsc">SNPAnnotationParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the observed alleles.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html#getOMRFBarCode--">getOMRFBarCode()</a></span> - Method in interface edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.omrf">TabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getOtherAllele--">getOtherAllele()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the other allele.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getOtherAlleleCount--">getOtherAlleleCount()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the number of other alleles observed.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getOtherAlleleFreq--">getOtherAlleleFreq()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the frequency of other allele.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html#getOtherDesignations--">getOtherDesignations()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">GeneInfoParser.Record</a></dt>
<dd>
<div class="block">Returns a pipe-delimited set of alternate descriptions that have been assigned to this gene.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html#getPAEndTiter--">getPAEndTiter()</a></span> - Method in interface edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.omrf">TabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getPanelLSID--">getPanelLSID()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the LSID for panel of individuals genotyped.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html#getPaternalID--">getPaternalID()</a></span> - Method in interface edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html" title="interface in edu.uab.ssg.io.plink">PEDParser.SampleRecord</a></dt>
<dd>
<div class="block">Returns the paternal ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html#getPhenotype--">getPhenotype()</a></span> - Method in interface edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html" title="interface in edu.uab.ssg.io.plink">PEDParser.SampleRecord</a></dt>
<dd>
<div class="block">Returns the phenotype.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html#getpos--">getpos()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">FullScoreFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the zero-based position of this SNP on the chromosome or 
 Integer.MIN_VALUE if the value is "NA" in the full score file.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getPos--">getPos()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the chromosome position on reference sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GTypeFileParser.SNPRecord.html#getPosition--">getPosition()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GTypeFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">GTypeFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the position of this SNP on the chromosome.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/SNPFileParser.SNPRecord.html#getPosition--">getPosition()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">SNPFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the position of this SNP on the chromosome.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html#getPosition--">getPosition()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.hapmap">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the position of the SNP on the chromosome.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.SNPRecord.html#getPosition--">getPosition()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.hapmap">HaplotypeFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the position of the SNP on the chromosome.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html#getPosition--">getPosition()</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.illumina">SNPFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the position of this SNP on the chromosome.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeRecord.html#getPosition--">getPosition()</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.marchini_gwas">GENSParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the position of the SNP on the chromosome.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html#getPosition--">getPosition()</a></span> - Method in interface edu.uab.ssg.io.the1000genomes.<a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html" title="interface in edu.uab.ssg.io.the1000genomes">VCFParser.VariantRecord</a></dt>
<dd>
<div class="block">Returns the position of this variant on the chromosome.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/DefaultSNP.html#getPosition--">getPosition()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/DefaultSNP.html" title="class in edu.uab.ssg.model.snp">DefaultSNP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/SNP.html#getPosition--">getPosition()</a></span> - Method in interface edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/SNP.html" title="interface in edu.uab.ssg.model.snp">SNP</a></dt>
<dd>
<div class="block">Returns the position on the chromosome where the SNP is located.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeProbabilities.html#getProbAA--">getProbAA()</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeProbabilities.html" title="interface in edu.uab.ssg.io.marchini_gwas">GENSParser.GenotypeProbabilities</a></dt>
<dd>
<div class="block">Returns the probability of genotype AA.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeProbabilities.html#getProbAB--">getProbAB()</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeProbabilities.html" title="interface in edu.uab.ssg.io.marchini_gwas">GENSParser.GenotypeProbabilities</a></dt>
<dd>
<div class="block">Returns the probability of genotype AB.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeProbabilities.html#getProbBB--">getProbBB()</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.GenotypeProbabilities.html" title="interface in edu.uab.ssg.io.marchini_gwas">GENSParser.GenotypeProbabilities</a></dt>
<dd>
<div class="block">Returns the probability of genotype BB.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getProtLSID--">getProtLSID()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the LSID for HapMap protocol used for genotyping.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getQC_code--">getQC_code()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the QC-code, currently 'QC+' for all entries (for future use).</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getRace--">getRace()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html#getRace--">getRace()</a></span> - Method in interface edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.omrf">TabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getRefAllele--">getRefAllele()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the reference allele.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getRefAlleleCount--">getRefAlleleCount()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the number of reference alleles observed.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getRefAlleleFreq--">getRefAlleleFreq()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the frequency of reference allele.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html#getReferenceAllele--">getReferenceAllele()</a></span> - Method in interface edu.uab.ssg.io.the1000genomes.<a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html" title="interface in edu.uab.ssg.io.the1000genomes">VCFParser.VariantRecord</a></dt>
<dd>
<div class="block">Returns the reference allele.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/AlleleCounter.html#getRelativeFrequency-java.lang.String-">getRelativeFrequency(String)</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/AlleleCounter.html" title="class in edu.uab.ssg.model.snp">AlleleCounter</a></dt>
<dd>
<div class="block">Returns the relative frequency of a particular allele within
 the study population.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getrfCHF--">getrfCHF()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getrfDM--">getrfDM()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getRsnum--">getRsnum()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the refSNP rs# identifier in dbSNP.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GenotypeFileParser.GenotypeRecord.html#getSampleID--">getSampleID()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.decode">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the sample identifier.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/fastphase/OUTParser.SampleRecord.html#getSampleID--">getSampleID()</a></span> - Method in interface edu.uab.ssg.io.fastphase.<a href="edu/uab/ssg/io/fastphase/OUTParser.SampleRecord.html" title="interface in edu.uab.ssg.io.fastphase">OUTParser.SampleRecord</a></dt>
<dd>
<div class="block">Returns the sample identifier.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html#getSampleID--">getSampleID()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.hapmap">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the sample identifier.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/GenotypeFileParser.GenotypeRecord.html#getSampleID--">getSampleID()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.limdi">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the sample identifier.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/SampleBuilder.html#getSampleName--">getSampleName()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/SampleBuilder.html" title="class in edu.uab.ssg.model.snp">SampleBuilder</a></dt>
<dd>
<div class="block">Returns the sample name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GTypeFileParser.SNPRecord.html#getSampleNames--">getSampleNames()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GTypeFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">GTypeFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the samples for which this record contains genotypes.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.SNPRecord.html#getSampleNames--">getSampleNames()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.hapmap">HaplotypeFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns a list of sample names for which there are haplotype data.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html#getSamples--">getSamples()</a></span> - Method in interface edu.uab.ssg.io.the1000genomes.<a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html" title="interface in edu.uab.ssg.io.the1000genomes">VCFParser.VariantRecord</a></dt>
<dd>
<div class="block">Returns the list of samples.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html#getSex--">getSex()</a></span> - Method in interface edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.omrf">TabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html#getSex--">getSex()</a></span> - Method in interface edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html" title="interface in edu.uab.ssg.io.plink">PEDParser.SampleRecord</a></dt>
<dd>
<div class="block">Returns the sex.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getsmoke--">getsmoke()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/SNPFileParser.SNPRecord.html#getSNP--">getSNP()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">SNPFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns a text representation of the nucleotides found at this 
 SNP position, for example, "[A/G]".</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html#getSNP--">getSNP()</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.illumina">SNPFileParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns a text representation of the nucleotides found at this 
 SNP position, for example, "[A/G]".</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/Sample.Genotype.html#getSNP--">getSNP()</a></span> - Method in interface edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/Sample.Genotype.html" title="interface in edu.uab.ssg.model.snp">Sample.Genotype</a></dt>
<dd>
<div class="block">Returns the SNP for this genotype.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html#getSNPClass--">getSNPClass()</a></span> - Method in interface edu.uab.ssg.io.ucsc.<a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html" title="interface in edu.uab.ssg.io.ucsc">SNPAnnotationParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the class (single, in-del, ..., insertion, deletion).</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html#getSNPName--">getSNPName()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">FullScoreFileParser.SNPRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GenotypeFileParser.GenotypeRecord.html#getSNPName--">getSNPName()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.decode">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the SNP name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html#getSNPName--">getSNPName()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.hapmap">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the SNP name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/GenotypeFileParser.GenotypeRecord.html#getSNPName--">getSNPName()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.limdi">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the SNP name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/Legend.html#getSNPs--">getSNPs()</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/Legend.html" title="interface in edu.uab.ssg.io.marchini_gwas">Legend</a></dt>
<dd>
<div class="block">Returns the SNPs for which this legend provides allele 0 and 1 recodes.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/LegendFile.html#getSNPs--">getSNPs()</a></span> - Method in class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/LegendFile.html" title="class in edu.uab.ssg.io.marchini_gwas">LegendFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getsqdose--">getsqdose()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/DoseTabParser.SampleRecord.html#getStabledose--">getStabledose()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/DoseTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">DoseTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getStatin--">getStatin()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getStrand--">getStrand()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the strand of reference sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html#getStrand--">getStrand()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.hapmap">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the strand.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html#getStrand--">getStrand()</a></span> - Method in interface edu.uab.ssg.io.ucsc.<a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html" title="interface in edu.uab.ssg.io.ucsc">SNPAnnotationParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the strand for the observed alleles.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/Sample.Genotype.html#getStrand--">getStrand()</a></span> - Method in interface edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/Sample.Genotype.html" title="interface in edu.uab.ssg.model.snp">Sample.Genotype</a></dt>
<dd>
<div class="block">Returns the strand of the alleles of this genotype.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/fastphase/OUTParser.SampleRecord.html#getSubpopLabel--">getSubpopLabel()</a></span> - Method in interface edu.uab.ssg.io.fastphase.<a href="edu/uab/ssg/io/fastphase/OUTParser.SampleRecord.html" title="interface in edu.uab.ssg.io.fastphase">OUTParser.SampleRecord</a></dt>
<dd>
<div class="block">Returns the subpopulation label.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html#getSymbol--">getSymbol()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">GeneInfoParser.Record</a></dt>
<dd>
<div class="block">Returns the default symbol for this gene.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html#getSymbolFromNomenclatureAuthority--">getSymbolFromNomenclatureAuthority()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">GeneInfoParser.Record</a></dt>
<dd>
<div class="block">Returns the symbol from nomenclature authority, if available.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html#getSynonyms--">getSynonyms()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">GeneInfoParser.Record</a></dt>
<dd>
<div class="block">Returns a pipe-delimited set of unofficial symbols or aliases for this gene.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html#getTaxID--">getTaxID()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">GeneInfoParser.Record</a></dt>
<dd>
<div class="block">Returns the unique identifier provided by the NCBI Taxonomy for the species or strain/isolate.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html#getTaxID--">getTaxID()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.Record</a></dt>
<dd>
<div class="block">Returns the unique identifier provided by the NCBI Taxonomy for the species or strain/isolate.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html#getTotalCount--">getTotalCount()</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.AlleleFreqRecord.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.AlleleFreqRecord</a></dt>
<dd>
<div class="block">Returns the total number of chromosomes observed.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html#getTranscript--">getTranscript()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">SeqGeneMdParser.Record</a></dt>
<dd>
<div class="block">Returns the transcript name (for RNA-related features).</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html#getTypeOfGene--">getTypeOfGene()</a></span> - Method in interface edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.Record.html" title="interface in edu.uab.ssg.io.ncbi">GeneInfoParser.Record</a></dt>
<dd>
<div class="block">Returns the type assigned to this gene according to the list of options provided in <a href="http://www.ncbi.nlm.nih.gov/IEB/ToolBox/CPP_DOC/lxr/source/src/objects/entrezgene/entrezgene.asn">http://www.ncbi.nlm.nih.gov/IEB/ToolBox/CPP_DOC/lxr/source/src/objects/entrezgene/entrezgene.asn</a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html#getValid--">getValid()</a></span> - Method in interface edu.uab.ssg.io.ucsc.<a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html" title="interface in edu.uab.ssg.io.ucsc">SNPAnnotationParser.SNPRecord</a></dt>
<dd>
<div class="block">Returns the validation strategy (by-cluster, ..., by-1000genomes).</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/GenotypeFileParser.GenotypeRecord.html#getValue-int-">getValue(int)</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.illumina">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the value for this record at the given column index.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/GenotypeFileParser.GenotypeRecord.html#getValue-int-java.lang.String-java.lang.String-">getValue(int, String, String)</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.illumina">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the value for this record at the given column index.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/GenotypeFileParser.GenotypeRecord.html#getValue-java.lang.String-">getValue(String)</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.illumina">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the value for this record corresponding to the given column name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/GenotypeFileParser.GenotypeRecord.html#getValue-java.lang.String-java.lang.String-java.lang.String-">getValue(String, String, String)</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.illumina">GenotypeFileParser.GenotypeRecord</a></dt>
<dd>
<div class="block">Returns the value for this record corresponding to the given column name.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html#getViability--">getViability()</a></span> - Method in interface edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.omrf">TabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getVKOR--">getVKOR()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html#getWeight--">getWeight()</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.SampleRecord</a></dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/decode/GTypeFileParser.html" title="class in edu.uab.ssg.io.decode"><span class="typeNameLink">GTypeFileParser</span></a> - Class in <a href="edu/uab/ssg/io/decode/package-summary.html">edu.uab.ssg.io.decode</a></dt>
<dd>
<div class="block">A parser for the <i>JHY_GType.txt</i> file format from deCODE containing A/B allele calls.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GTypeFileParser.html#GTypeFileParser--">GTypeFileParser()</a></span> - Constructor for class edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GTypeFileParser.html" title="class in edu.uab.ssg.io.decode">GTypeFileParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/decode/GTypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.decode"><span class="typeNameLink">GTypeFileParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/decode/package-summary.html">edu.uab.ssg.io.decode</a></dt>
<dd>
<div class="block">A listener for handling parsed SNP records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/decode/GTypeFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode"><span class="typeNameLink">GTypeFileParser.SNPRecord</span></a> - Interface in <a href="edu/uab/ssg/io/decode/package-summary.html">edu.uab.ssg.io.decode</a></dt>
<dd>
<div class="block">A SNP record.</div>
</dd>
</dl>
<a name="I:H">
<!--   -->
</a>
<h2 class="title">H</h2>
<dl>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.decode">FullScoreFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GenotypeFileParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GenotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.decode">GenotypeFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GTypeFileParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GTypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.decode">GTypeFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/SNPFileParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/SNPFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.decode">SNPFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/fastphase/OUTParser.RecordListener.html#handleBadRecordFormat-java.lang.String-java.lang.String-java.lang.String-">handleBadRecordFormat(String, String, String)</a></span> - Method in interface edu.uab.ssg.io.fastphase.<a href="edu/uab/ssg/io/fastphase/OUTParser.RecordListener.html" title="interface in edu.uab.ssg.io.fastphase">OUTParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.hapmap">GenotypeFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.hapmap">HaplotypeFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/GenotypeFileParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/GenotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.illumina">GenotypeFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/SNPFileParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/SNPFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.illumina">SNPFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.RecordListener.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/DoseTabParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/DoseTabParser.RecordListener.html" title="interface in edu.uab.ssg.io.limdi">DoseTabParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/GenotypeFileParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/GenotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.limdi">GenotypeFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/mach/GENOParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.mach.<a href="edu/uab/ssg/io/mach/GENOParser.RecordListener.html" title="interface in edu.uab.ssg.io.mach">GENOParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.RecordListener.html" title="interface in edu.uab.ssg.io.marchini_gwas">GENSParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.RecordListener.html" title="interface in edu.uab.ssg.io.omrf">TabParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/BIMParser.BadRecordFormatListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/BIMParser.BadRecordFormatListener.html" title="interface in edu.uab.ssg.io.plink">BIMParser.BadRecordFormatListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/MAPParser.BadRecordFormatListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/MAPParser.BadRecordFormatListener.html" title="interface in edu.uab.ssg.io.plink">MAPParser.BadRecordFormatListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDParser.RecordListener.html" title="interface in edu.uab.ssg.io.plink">PEDParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/the1000genomes/VCFParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.the1000genomes.<a href="edu/uab/ssg/io/the1000genomes/VCFParser.RecordListener.html" title="interface in edu.uab.ssg.io.the1000genomes">VCFParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.RecordListener.html#handleBadRecordFormat-java.lang.String-">handleBadRecordFormat(String)</a></span> - Method in interface edu.uab.ssg.io.ucsc.<a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.RecordListener.html" title="interface in edu.uab.ssg.io.ucsc">SNPAnnotationParser.RecordListener</a></dt>
<dd>
<div class="block">Handles input that could not be parsed due to a formatting problem.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.decode.FullScoreFileParser.SNPRecord-">handleParsedRecord(FullScoreFileParser.SNPRecord)</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.decode">FullScoreFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed SNP record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GenotypeFileParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.decode.GenotypeFileParser.GenotypeRecord-">handleParsedRecord(GenotypeFileParser.GenotypeRecord)</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GenotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.decode">GenotypeFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed genotype record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GTypeFileParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.decode.GTypeFileParser.SNPRecord-">handleParsedRecord(GTypeFileParser.SNPRecord)</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GTypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.decode">GTypeFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed SNP record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/SNPFileParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.decode.SNPFileParser.SNPRecord-">handleParsedRecord(SNPFileParser.SNPRecord)</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/SNPFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.decode">SNPFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed SNP record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/fastphase/OUTParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.fastphase.OUTParser.SampleRecord-">handleParsedRecord(OUTParser.SampleRecord)</a></span> - Method in interface edu.uab.ssg.io.fastphase.<a href="edu/uab/ssg/io/fastphase/OUTParser.RecordListener.html" title="interface in edu.uab.ssg.io.fastphase">OUTParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed sample record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.hapmap.AlleleFreqFileParser.AlleleFreqRecord-">handleParsedRecord(AlleleFreqFileParser.AlleleFreqRecord)</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.hapmap">AlleleFreqFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed allele frequency record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.hapmap.GenotypeFileParser.GenotypeRecord-">handleParsedRecord(GenotypeFileParser.GenotypeRecord)</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.hapmap">GenotypeFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed genotype record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.hapmap.HaplotypeFileParser.SNPRecord-">handleParsedRecord(HaplotypeFileParser.SNPRecord)</a></span> - Method in interface edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.hapmap">HaplotypeFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed SNP record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/GenotypeFileParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.illumina.GenotypeFileParser.GenotypeRecord-">handleParsedRecord(GenotypeFileParser.GenotypeRecord)</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/GenotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.illumina">GenotypeFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed genotype record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/SNPFileParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.illumina.SNPFileParser.SNPRecord-">handleParsedRecord(SNPFileParser.SNPRecord)</a></span> - Method in interface edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/SNPFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.illumina">SNPFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed SNP record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.limdi.CovariateTabParser.SampleRecord-">handleParsedRecord(CovariateTabParser.SampleRecord)</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.RecordListener.html" title="interface in edu.uab.ssg.io.limdi">CovariateTabParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed sample record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/DoseTabParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.limdi.DoseTabParser.SampleRecord-">handleParsedRecord(DoseTabParser.SampleRecord)</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/DoseTabParser.RecordListener.html" title="interface in edu.uab.ssg.io.limdi">DoseTabParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed sample record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/GenotypeFileParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.limdi.GenotypeFileParser.GenotypeRecord-">handleParsedRecord(GenotypeFileParser.GenotypeRecord)</a></span> - Method in interface edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/GenotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.limdi">GenotypeFileParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed genotype record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/mach/GENOParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.mach.GENOParser.IndividualRecord-">handleParsedRecord(GENOParser.IndividualRecord)</a></span> - Method in interface edu.uab.ssg.io.mach.<a href="edu/uab/ssg/io/mach/GENOParser.RecordListener.html" title="interface in edu.uab.ssg.io.mach">GENOParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed individual record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.marchini_gwas.GENSParser.GenotypeRecord-">handleParsedRecord(GENSParser.GenotypeRecord)</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.RecordListener.html" title="interface in edu.uab.ssg.io.marchini_gwas">GENSParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed genotype record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.omrf.TabParser.SampleRecord-">handleParsedRecord(TabParser.SampleRecord)</a></span> - Method in interface edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.RecordListener.html" title="interface in edu.uab.ssg.io.omrf">TabParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed sample record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.plink.PEDParser.SampleRecord-">handleParsedRecord(PEDParser.SampleRecord)</a></span> - Method in interface edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDParser.RecordListener.html" title="interface in edu.uab.ssg.io.plink">PEDParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed sample record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/the1000genomes/VCFParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.the1000genomes.VCFParser.VariantRecord-">handleParsedRecord(VCFParser.VariantRecord)</a></span> - Method in interface edu.uab.ssg.io.the1000genomes.<a href="edu/uab/ssg/io/the1000genomes/VCFParser.RecordListener.html" title="interface in edu.uab.ssg.io.the1000genomes">VCFParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed variant record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.RecordListener.html#handleParsedRecord-edu.uab.ssg.io.ucsc.SNPAnnotationParser.SNPRecord-">handleParsedRecord(SNPAnnotationParser.SNPRecord)</a></span> - Method in interface edu.uab.ssg.io.ucsc.<a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.RecordListener.html" title="interface in edu.uab.ssg.io.ucsc">SNPAnnotationParser.RecordListener</a></dt>
<dd>
<div class="block">Handles successfully parsed SNP annotation record.</div>
</dd>
<dt><a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.html" title="class in edu.uab.ssg.io.hapmap"><span class="typeNameLink">HaplotypeFileParser</span></a> - Class in <a href="edu/uab/ssg/io/hapmap/package-summary.html">edu.uab.ssg.io.hapmap</a></dt>
<dd>
<div class="block">A parser for the "snps x haplotype" file format for the HapMap phase3 data located at <a href="http://ftp.hapmap.org/phasing/2009-02_phaseIII/HapMap3_r2/">http://ftp.hapmap.org/phasing/2009-02_phaseIII/HapMap3_r2/</a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.html#HaplotypeFileParser--">HaplotypeFileParser()</a></span> - Constructor for class edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.html" title="class in edu.uab.ssg.io.hapmap">HaplotypeFileParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.hapmap"><span class="typeNameLink">HaplotypeFileParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/hapmap/package-summary.html">edu.uab.ssg.io.hapmap</a></dt>
<dd>
<div class="block">A listener for handling parsed SNP records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.hapmap"><span class="typeNameLink">HaplotypeFileParser.SNPRecord</span></a> - Interface in <a href="edu/uab/ssg/io/hapmap/package-summary.html">edu.uab.ssg.io.hapmap</a></dt>
<dd>
<div class="block">A "SNP x haplotype" record.</div>
</dd>
<dt><a href="edu/uab/ssg/io/marchini_gwas/HAPSWriter.html" title="class in edu.uab.ssg.io.marchini_gwas"><span class="typeNameLink">HAPSWriter</span></a> - Class in <a href="edu/uab/ssg/io/marchini_gwas/package-summary.html">edu.uab.ssg.io.marchini_gwas</a></dt>
<dd>
<div class="block">A writer for the HAPS file format of the HapGen software at <a href="http://www.stats.ox.ac.uk/~marchini/software/gwas/hapgen.html">http://www.stats.ox.ac.uk/~marchini/software/gwas/hapgen.html</a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/HAPSWriter.html#HAPSWriter--">HAPSWriter()</a></span> - Constructor for class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/HAPSWriter.html" title="class in edu.uab.ssg.io.marchini_gwas">HAPSWriter</a></dt>
<dd>
<div class="block">Constructs the writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/DefaultSNP.html#hashCode--">hashCode()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/DefaultSNP.html" title="class in edu.uab.ssg.model.snp">DefaultSNP</a></dt>
<dd>
<div class="block">Returns the hash code.</div>
</dd>
</dl>
<a name="I:I">
<!--   -->
</a>
<h2 class="title">I</h2>
<dl>
<dt><a href="edu/uab/ssg/model/snp/IlluminaStrand.html" title="class in edu.uab.ssg.model.snp"><span class="typeNameLink">IlluminaStrand</span></a> - Class in <a href="edu/uab/ssg/model/snp/package-summary.html">edu.uab.ssg.model.snp</a></dt>
<dd>
<div class="block">A strand designation, developed by Illumina, "based on the actual or
 contextual sequence of each individual SNP".</div>
</dd>
<dt><a href="edu/uab/ssg/io/phase/INPWriter.html" title="class in edu.uab.ssg.io.phase"><span class="typeNameLink">INPWriter</span></a> - Class in <a href="edu/uab/ssg/io/phase/package-summary.html">edu.uab.ssg.io.phase</a></dt>
<dd>
<div class="block">A writer for the PHASE INP input file format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/phase/INPWriter.html#INPWriter--">INPWriter()</a></span> - Constructor for class edu.uab.ssg.io.phase.<a href="edu/uab/ssg/io/phase/INPWriter.html" title="class in edu.uab.ssg.io.phase">INPWriter</a></dt>
<dd>
<div class="block">Constructs the writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html#isAIM--">isAIM()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">FullScoreFileParser.SNPRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/AlleleCounter.html#isAmbiguous--">isAmbiguous()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/AlleleCounter.html" title="class in edu.uab.ssg.model.snp">AlleleCounter</a></dt>
<dd>
<div class="block">Returns true if counted alleles indicate an ambiguous A/T or G/C SNP.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/AlleleCounter.html#isBiallelic--">isBiallelic()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/AlleleCounter.html" title="class in edu.uab.ssg.model.snp">AlleleCounter</a></dt>
<dd>
<div class="block">Returns true if the marker is bi-allelic.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html#isCNV--">isCNV()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">FullScoreFileParser.SNPRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html#isHarley--">isHarley()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">FullScoreFileParser.SNPRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html#isInclude--">isInclude()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">FullScoreFileParser.SNPRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html#isMHC--">isMHC()</a></span> - Method in interface edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode">FullScoreFileParser.SNPRecord</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/AlleleCounter.html#isMonomorphic--">isMonomorphic()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/AlleleCounter.html" title="class in edu.uab.ssg.model.snp">AlleleCounter</a></dt>
<dd>
<div class="block">Returns true if the marker is monomorphic.</div>
</dd>
</dl>
<a name="I:L">
<!--   -->
</a>
<h2 class="title">L</h2>
<dl>
<dt><a href="edu/uab/ssg/io/marchini_gwas/Legend.html" title="interface in edu.uab.ssg.io.marchini_gwas"><span class="typeNameLink">Legend</span></a> - Interface in <a href="edu/uab/ssg/io/marchini_gwas/package-summary.html">edu.uab.ssg.io.marchini_gwas</a></dt>
<dd>
<div class="block">An abstraction for the "legend" concept and file format required by the HapGen software at <a href="http://www.stats.ox.ac.uk/~marchini/software/gwas/hapgen.html">http://www.stats.ox.ac.uk/~marchini/software/gwas/hapgen.html</a>.</div>
</dd>
<dt><a href="edu/uab/ssg/io/marchini_gwas/LegendBuilder.html" title="class in edu.uab.ssg.io.marchini_gwas"><span class="typeNameLink">LegendBuilder</span></a> - Class in <a href="edu/uab/ssg/io/marchini_gwas/package-summary.html">edu.uab.ssg.io.marchini_gwas</a></dt>
<dd>
<div class="block">A builder for Legend implementations.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/LegendBuilder.html#LegendBuilder--">LegendBuilder()</a></span> - Constructor for class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/LegendBuilder.html" title="class in edu.uab.ssg.io.marchini_gwas">LegendBuilder</a></dt>
<dd>
<div class="block">Constructs the builder.</div>
</dd>
<dt><a href="edu/uab/ssg/io/marchini_gwas/LegendFile.html" title="class in edu.uab.ssg.io.marchini_gwas"><span class="typeNameLink">LegendFile</span></a> - Class in <a href="edu/uab/ssg/io/marchini_gwas/package-summary.html">edu.uab.ssg.io.marchini_gwas</a></dt>
<dd>
<div class="block">A file-based Legend implementation.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/LegendFile.html#LegendFile-java.io.InputStream-">LegendFile(InputStream)</a></span> - Constructor for class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/LegendFile.html" title="class in edu.uab.ssg.io.marchini_gwas">LegendFile</a></dt>
<dd>
<div class="block">Constructs the legend from a user-supplied input stream.</div>
</dd>
</dl>
<a name="I:M">
<!--   -->
</a>
<h2 class="title">M</h2>
<dl>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/reports/FlipCalls.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class edu.uab.ssg.reports.<a href="edu/uab/ssg/reports/FlipCalls.html" title="class in edu.uab.ssg.reports">FlipCalls</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/reports/GeneReport.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class edu.uab.ssg.reports.<a href="edu/uab/ssg/reports/GeneReport.html" title="class in edu.uab.ssg.reports">GeneReport</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/reports/MarkerAnnotationReport.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class edu.uab.ssg.reports.<a href="edu/uab/ssg/reports/MarkerAnnotationReport.html" title="class in edu.uab.ssg.reports">MarkerAnnotationReport</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/reports/StrandMatchReport.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class edu.uab.ssg.reports.<a href="edu/uab/ssg/reports/StrandMatchReport.html" title="class in edu.uab.ssg.reports">StrandMatchReport</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/Sex.html#MALE">MALE</a></span> - Static variable in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/Sex.html" title="class in edu.uab.ssg.model.snp">Sex</a></dt>
<dd>
<div class="block">The male sex.</div>
</dd>
<dt><a href="edu/uab/ssg/io/plink/MAPParser.html" title="class in edu.uab.ssg.io.plink"><span class="typeNameLink">MAPParser</span></a> - Class in <a href="edu/uab/ssg/io/plink/package-summary.html">edu.uab.ssg.io.plink</a></dt>
<dd>
<div class="block">A parser for the PLINK MAP input file format described in the <a href="http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml">basic usage/data formats section</a> of the online documentation.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/MAPParser.html#MAPParser--">MAPParser()</a></span> - Constructor for class edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/MAPParser.html" title="class in edu.uab.ssg.io.plink">MAPParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/plink/MAPParser.BadRecordFormatListener.html" title="interface in edu.uab.ssg.io.plink"><span class="typeNameLink">MAPParser.BadRecordFormatListener</span></a> - Interface in <a href="edu/uab/ssg/io/plink/package-summary.html">edu.uab.ssg.io.plink</a></dt>
<dd>
<div class="block">A listener for handling problems due to bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/plink/MAPWriter.html" title="class in edu.uab.ssg.io.plink"><span class="typeNameLink">MAPWriter</span></a> - Class in <a href="edu/uab/ssg/io/plink/package-summary.html">edu.uab.ssg.io.plink</a></dt>
<dd>
<div class="block">A writer for the PLINK MAP input file format described in the <a href="http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml">basic usage/data formats section</a> of the online documentation.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/MAPWriter.html#MAPWriter--">MAPWriter()</a></span> - Constructor for class edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/MAPWriter.html" title="class in edu.uab.ssg.io.plink">MAPWriter</a></dt>
<dd>
<div class="block">Constructs the writer.</div>
</dd>
<dt><a href="edu/uab/ssg/reports/MarkerAnnotationReport.html" title="class in edu.uab.ssg.reports"><span class="typeNameLink">MarkerAnnotationReport</span></a> - Class in <a href="edu/uab/ssg/reports/package-summary.html">edu.uab.ssg.reports</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/reports/MarkerAnnotationReport.html#MarkerAnnotationReport--">MarkerAnnotationReport()</a></span> - Constructor for class edu.uab.ssg.reports.<a href="edu/uab/ssg/reports/MarkerAnnotationReport.html" title="class in edu.uab.ssg.reports">MarkerAnnotationReport</a></dt>
<dd>&nbsp;</dd>
<dt><a href="edu/uab/ssg/io/haploview/MarkerInfoWriter.html" title="class in edu.uab.ssg.io.haploview"><span class="typeNameLink">MarkerInfoWriter</span></a> - Class in <a href="edu/uab/ssg/io/haploview/package-summary.html">edu.uab.ssg.io.haploview</a></dt>
<dd>
<div class="block">A writer for the Haploview marker information input file format described in the <a href="http://www.broadinstitute.org/science/programs/medical-and-population-genetics/haploview/input-file-formats-0">Input File Formats section</a> of the Files chapter of the <a href="http://www.broadinstitute.org/haploview/user-manual">online user manual</a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/haploview/MarkerInfoWriter.html#MarkerInfoWriter--">MarkerInfoWriter()</a></span> - Constructor for class edu.uab.ssg.io.haploview.<a href="edu/uab/ssg/io/haploview/MarkerInfoWriter.html" title="class in edu.uab.ssg.io.haploview">MarkerInfoWriter</a></dt>
<dd>
<div class="block">Constructs the writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/GenotypeFileParser.html#MISSING_ALLELE">MISSING_ALLELE</a></span> - Static variable in class edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/GenotypeFileParser.html" title="class in edu.uab.ssg.io.illumina">GenotypeFileParser</a></dt>
<dd>
<div class="block">A string constant for the character (a dash, '-') that represents a missing allele call in the Illumina genotyping report format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDParser.html#MISSING_VALUE">MISSING_VALUE</a></span> - Static variable in class edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDParser.html" title="class in edu.uab.ssg.io.plink">PEDParser</a></dt>
<dd>
<div class="block">The character in this file format that represents a missing value.</div>
</dd>
</dl>
<a name="I:O">
<!--   -->
</a>
<h2 class="title">O</h2>
<dl>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/Util.html#ORDER_BY_CHROMOSOME_POSITION">ORDER_BY_CHROMOSOME_POSITION</a></span> - Static variable in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/Util.html" title="class in edu.uab.ssg.model.snp">Util</a></dt>
<dd>
<div class="block">A Comparator that orders <tt>SNP</tt> objects by chromosome (lexicographically by name) and position.</div>
</dd>
<dt><a href="edu/uab/ssg/io/fastphase/OUTParser.html" title="class in edu.uab.ssg.io.fastphase"><span class="typeNameLink">OUTParser</span></a> - Class in <a href="edu/uab/ssg/io/fastphase/package-summary.html">edu.uab.ssg.io.fastphase</a></dt>
<dd>
<div class="block">A parser for the fastPHASE inferred haplotypes output file format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/fastphase/OUTParser.html#OUTParser--">OUTParser()</a></span> - Constructor for class edu.uab.ssg.io.fastphase.<a href="edu/uab/ssg/io/fastphase/OUTParser.html" title="class in edu.uab.ssg.io.fastphase">OUTParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/fastphase/OUTParser.RecordListener.html" title="interface in edu.uab.ssg.io.fastphase"><span class="typeNameLink">OUTParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/fastphase/package-summary.html">edu.uab.ssg.io.fastphase</a></dt>
<dd>
<div class="block">A listener for handling parsed sample records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/fastphase/OUTParser.SampleRecord.html" title="interface in edu.uab.ssg.io.fastphase"><span class="typeNameLink">OUTParser.SampleRecord</span></a> - Interface in <a href="edu/uab/ssg/io/fastphase/package-summary.html">edu.uab.ssg.io.fastphase</a></dt>
<dd>
<div class="block">A sample record.</div>
</dd>
<dt><a href="edu/uab/ssg/io/fastphase/OUTWriter.html" title="class in edu.uab.ssg.io.fastphase"><span class="typeNameLink">OUTWriter</span></a> - Class in <a href="edu/uab/ssg/io/fastphase/package-summary.html">edu.uab.ssg.io.fastphase</a></dt>
<dd>
<div class="block">A writer for the fastPHASE inferred haplotypes output file format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/fastphase/OUTWriter.html#OUTWriter-java.util.List-java.io.OutputStream-">OUTWriter(List&lt;SNP&gt;, OutputStream)</a></span> - Constructor for class edu.uab.ssg.io.fastphase.<a href="edu/uab/ssg/io/fastphase/OUTWriter.html" title="class in edu.uab.ssg.io.fastphase">OUTWriter</a></dt>
<dd>
<div class="block">Constructs the writer.</div>
</dd>
</dl>
<a name="I:P">
<!--   -->
</a>
<h2 class="title">P</h2>
<dl>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/FullScoreFileParser.html#parse-java.io.InputStream-edu.uab.ssg.io.decode.FullScoreFileParser.RecordListener-">parse(InputStream, FullScoreFileParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/FullScoreFileParser.html" title="class in edu.uab.ssg.io.decode">FullScoreFileParser</a></dt>
<dd>
<div class="block">Parses the input stream for SNP records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GenotypeFileParser.html#parse-java.io.InputStream-edu.uab.ssg.io.decode.GenotypeFileParser.RecordListener-">parse(InputStream, GenotypeFileParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GenotypeFileParser.html" title="class in edu.uab.ssg.io.decode">GenotypeFileParser</a></dt>
<dd>
<div class="block">Parses the input stream for genotype records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/GTypeFileParser.html#parse-java.io.InputStream-edu.uab.ssg.io.decode.GTypeFileParser.RecordListener-">parse(InputStream, GTypeFileParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/GTypeFileParser.html" title="class in edu.uab.ssg.io.decode">GTypeFileParser</a></dt>
<dd>
<div class="block">Parses the input stream for SNP records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/SNPFileParser.html#parse-java.io.InputStream-edu.uab.ssg.io.decode.SNPFileParser.RecordListener-">parse(InputStream, SNPFileParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/SNPFileParser.html" title="class in edu.uab.ssg.io.decode">SNPFileParser</a></dt>
<dd>
<div class="block">Parses the input stream for SNP records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/fastphase/OUTParser.html#parse-java.io.InputStream-edu.uab.ssg.io.fastphase.OUTParser.RecordListener-">parse(InputStream, OUTParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.fastphase.<a href="edu/uab/ssg/io/fastphase/OUTParser.html" title="class in edu.uab.ssg.io.fastphase">OUTParser</a></dt>
<dd>
<div class="block">Parses the input stream for sample records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.html#parse-java.io.InputStream-edu.uab.ssg.io.hapmap.AlleleFreqFileParser.RecordListener-">parse(InputStream, AlleleFreqFileParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/AlleleFreqFileParser.html" title="class in edu.uab.ssg.io.hapmap">AlleleFreqFileParser</a></dt>
<dd>
<div class="block">Parses the input stream for allele frequency records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.html#parse-java.io.InputStream-edu.uab.ssg.io.hapmap.GenotypeFileParser.RecordListener-">parse(InputStream, GenotypeFileParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/GenotypeFileParser.html" title="class in edu.uab.ssg.io.hapmap">GenotypeFileParser</a></dt>
<dd>
<div class="block">Parses the input stream for genotype records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.html#parse-java.io.InputStream-edu.uab.ssg.io.hapmap.HaplotypeFileParser.RecordListener-">parse(InputStream, HaplotypeFileParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.hapmap.<a href="edu/uab/ssg/io/hapmap/HaplotypeFileParser.html" title="class in edu.uab.ssg.io.hapmap">HaplotypeFileParser</a></dt>
<dd>
<div class="block">Parses the input stream for "SNP x haplotype" records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/GenotypeFileParser.html#parse-java.io.InputStream-edu.uab.ssg.io.illumina.GenotypeFileParser.RecordListener-">parse(InputStream, GenotypeFileParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/GenotypeFileParser.html" title="class in edu.uab.ssg.io.illumina">GenotypeFileParser</a></dt>
<dd>
<div class="block">Parses the input stream for genotype records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/SNPFileParser.html#parse-java.io.InputStream-edu.uab.ssg.io.illumina.SNPFileParser.RecordListener-">parse(InputStream, SNPFileParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/SNPFileParser.html" title="class in edu.uab.ssg.io.illumina">SNPFileParser</a></dt>
<dd>
<div class="block">Parses the input stream for SNP records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/CovariateTabParser.html#parse-java.io.InputStream-edu.uab.ssg.io.limdi.CovariateTabParser.RecordListener-">parse(InputStream, CovariateTabParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/CovariateTabParser.html" title="class in edu.uab.ssg.io.limdi">CovariateTabParser</a></dt>
<dd>
<div class="block">Parses the input stream for sample records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/DoseTabParser.html#parse-java.io.InputStream-edu.uab.ssg.io.limdi.DoseTabParser.RecordListener-">parse(InputStream, DoseTabParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/DoseTabParser.html" title="class in edu.uab.ssg.io.limdi">DoseTabParser</a></dt>
<dd>
<div class="block">Parses the input stream for sample records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/limdi/GenotypeFileParser.html#parse-java.io.InputStream-edu.uab.ssg.io.limdi.GenotypeFileParser.RecordListener-">parse(InputStream, GenotypeFileParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.limdi.<a href="edu/uab/ssg/io/limdi/GenotypeFileParser.html" title="class in edu.uab.ssg.io.limdi">GenotypeFileParser</a></dt>
<dd>
<div class="block">Parses the input stream for genotype records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/mach/GENOParser.html#parse-java.io.InputStream-edu.uab.ssg.io.mach.GENOParser.RecordListener-">parse(InputStream, GENOParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.mach.<a href="edu/uab/ssg/io/mach/GENOParser.html" title="class in edu.uab.ssg.io.mach">GENOParser</a></dt>
<dd>
<div class="block">Parses the input stream for individual records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSParser.html#parse-java.io.InputStream-edu.uab.ssg.io.marchini_gwas.GENSParser.RecordListener-">parse(InputStream, GENSParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSParser.html" title="class in edu.uab.ssg.io.marchini_gwas">GENSParser</a></dt>
<dd>
<div class="block">Parses the input stream for genotype records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/GeneInfoParser.html#parse-java.io.InputStream-">parse(InputStream)</a></span> - Method in class edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/GeneInfoParser.html" title="class in edu.uab.ssg.io.ncbi">GeneInfoParser</a></dt>
<dd>
<div class="block">Parses the input stream for gene info records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.html#parse-java.io.InputStream-">parse(InputStream)</a></span> - Method in class edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.html" title="class in edu.uab.ssg.io.ncbi">SeqGeneMdParser</a></dt>
<dd>
<div class="block">Parses the input stream for gene info records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.html#parse-java.io.InputStream-edu.uab.ssg.io.ncbi.SeqGeneMdParser.RecordFilter-">parse(InputStream, SeqGeneMdParser.RecordFilter)</a></span> - Method in class edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.html" title="class in edu.uab.ssg.io.ncbi">SeqGeneMdParser</a></dt>
<dd>
<div class="block">Parses the input stream for gene info records that meet the filtering criteria.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.html#parse-java.io.InputStream-edu.uab.ssg.io.omrf.TabParser.RecordListener-">parse(InputStream, TabParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.html" title="class in edu.uab.ssg.io.omrf">TabParser</a></dt>
<dd>
<div class="block">Parses the input stream for sample records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/BIMParser.html#parse-java.io.InputStream-edu.uab.ssg.io.plink.BIMParser.BadRecordFormatListener-">parse(InputStream, BIMParser.BadRecordFormatListener)</a></span> - Method in class edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/BIMParser.html" title="class in edu.uab.ssg.io.plink">BIMParser</a></dt>
<dd>
<div class="block">Parses the given input stream for SNP records in PLINK extended MAP file format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/MAPParser.html#parse-java.io.InputStream-edu.uab.ssg.io.plink.MAPParser.BadRecordFormatListener-">parse(InputStream, MAPParser.BadRecordFormatListener)</a></span> - Method in class edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/MAPParser.html" title="class in edu.uab.ssg.io.plink">MAPParser</a></dt>
<dd>
<div class="block">Parses the given input stream for SNP records in PLINK MAP file format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDParser.html#parse-java.io.InputStream-edu.uab.ssg.io.plink.PEDParser.RecordListener-">parse(InputStream, PEDParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDParser.html" title="class in edu.uab.ssg.io.plink">PEDParser</a></dt>
<dd>
<div class="block">Parses the input stream for sample records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/the1000genomes/VCFParser.html#parse-java.io.InputStream-edu.uab.ssg.io.the1000genomes.VCFParser.RecordListener-">parse(InputStream, VCFParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.the1000genomes.<a href="edu/uab/ssg/io/the1000genomes/VCFParser.html" title="class in edu.uab.ssg.io.the1000genomes">VCFParser</a></dt>
<dd>
<div class="block">Parses the input stream for metadata and variant records.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.html#parse-java.io.InputStream-edu.uab.ssg.io.ucsc.SNPAnnotationParser.RecordListener-">parse(InputStream, SNPAnnotationParser.RecordListener)</a></span> - Method in class edu.uab.ssg.io.ucsc.<a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.html" title="class in edu.uab.ssg.io.ucsc">SNPAnnotationParser</a></dt>
<dd>
<div class="block">Parses the input stream for SNP annotation records.</div>
</dd>
<dt><a href="edu/uab/ssg/io/plink/PEDParser.html" title="class in edu.uab.ssg.io.plink"><span class="typeNameLink">PEDParser</span></a> - Class in <a href="edu/uab/ssg/io/plink/package-summary.html">edu.uab.ssg.io.plink</a></dt>
<dd>
<div class="block">A parser for the PLINK PED input file format described in <a href="http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml">the basic usage/data formats section</a> of the online documentation.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDParser.html#PEDParser--">PEDParser()</a></span> - Constructor for class edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDParser.html" title="class in edu.uab.ssg.io.plink">PEDParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/plink/PEDParser.RecordListener.html" title="interface in edu.uab.ssg.io.plink"><span class="typeNameLink">PEDParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/plink/package-summary.html">edu.uab.ssg.io.plink</a></dt>
<dd>
<div class="block">A listener for handling parsed sample records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/plink/PEDParser.SampleRecord.html" title="interface in edu.uab.ssg.io.plink"><span class="typeNameLink">PEDParser.SampleRecord</span></a> - Interface in <a href="edu/uab/ssg/io/plink/package-summary.html">edu.uab.ssg.io.plink</a></dt>
<dd>
<div class="block">A sample record.</div>
</dd>
<dt><a href="edu/uab/ssg/io/plink/PEDWriter.html" title="class in edu.uab.ssg.io.plink"><span class="typeNameLink">PEDWriter</span></a> - Class in <a href="edu/uab/ssg/io/plink/package-summary.html">edu.uab.ssg.io.plink</a></dt>
<dd>
<div class="block">A writer for the PLINK PED input file format described in <a href="http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml">the basic usage/data formats section</a> of the online documentation.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDWriter.html#PEDWriter-java.util.List-java.io.OutputStream-">PEDWriter(List&lt;SNP&gt;, OutputStream)</a></span> - Constructor for class edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDWriter.html" title="class in edu.uab.ssg.io.plink">PEDWriter</a></dt>
<dd>
<div class="block">Constructs the writer.</div>
</dd>
</dl>
<a name="I:S">
<!--   -->
</a>
<h2 class="title">S</h2>
<dl>
<dt><a href="edu/uab/ssg/model/snp/Sample.html" title="interface in edu.uab.ssg.model.snp"><span class="typeNameLink">Sample</span></a> - Interface in <a href="edu/uab/ssg/model/snp/package-summary.html">edu.uab.ssg.model.snp</a></dt>
<dd>
<div class="block">A sample from a study population.</div>
</dd>
<dt><a href="edu/uab/ssg/model/snp/Sample.Genotype.html" title="interface in edu.uab.ssg.model.snp"><span class="typeNameLink">Sample.Genotype</span></a> - Interface in <a href="edu/uab/ssg/model/snp/package-summary.html">edu.uab.ssg.model.snp</a></dt>
<dd>
<div class="block">A genotype for a sample at a particular SNP.</div>
</dd>
<dt><a href="edu/uab/ssg/model/snp/SampleBuilder.html" title="class in edu.uab.ssg.model.snp"><span class="typeNameLink">SampleBuilder</span></a> - Class in <a href="edu/uab/ssg/model/snp/package-summary.html">edu.uab.ssg.model.snp</a></dt>
<dd>
<div class="block">A builder for a sample from a study population.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/SampleBuilder.html#SampleBuilder-java.lang.String-">SampleBuilder(String)</a></span> - Constructor for class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/SampleBuilder.html" title="class in edu.uab.ssg.model.snp">SampleBuilder</a></dt>
<dd>
<div class="block">Constructs a builder for a sample with the given name.</div>
</dd>
<dt><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.html" title="class in edu.uab.ssg.io.ncbi"><span class="typeNameLink">SeqGeneMdParser</span></a> - Class in <a href="edu/uab/ssg/io/ncbi/package-summary.html">edu.uab.ssg.io.ncbi</a></dt>
<dd>
<div class="block">A parser for the Entrez Gene seq_gene.md file located at <a href="./edu/uab/ssg/io/ncbi/ftp://ftp.ncbi.nih.gov/genomes/H_sapiens/mapview/seq_gene.md.gz">ftp://ftp.ncbi.nih.gov/genomes/H_sapiens/mapview/seq_gene.md.gz</a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.html#SeqGeneMdParser--">SeqGeneMdParser()</a></span> - Constructor for class edu.uab.ssg.io.ncbi.<a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.html" title="class in edu.uab.ssg.io.ncbi">SeqGeneMdParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.Record.html" title="interface in edu.uab.ssg.io.ncbi"><span class="typeNameLink">SeqGeneMdParser.Record</span></a> - Interface in <a href="edu/uab/ssg/io/ncbi/package-summary.html">edu.uab.ssg.io.ncbi</a></dt>
<dd>
<div class="block">A gene info record.</div>
</dd>
<dt><a href="edu/uab/ssg/io/ncbi/SeqGeneMdParser.RecordFilter.html" title="interface in edu.uab.ssg.io.ncbi"><span class="typeNameLink">SeqGeneMdParser.RecordFilter</span></a> - Interface in <a href="edu/uab/ssg/io/ncbi/package-summary.html">edu.uab.ssg.io.ncbi</a></dt>
<dd>
<div class="block">A record filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/SampleBuilder.html#setGenotype-edu.uab.ssg.model.snp.SNP-java.lang.String-java.lang.String-edu.uab.ssg.model.snp.Strand-">setGenotype(SNP, String, String, Strand)</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/SampleBuilder.html" title="class in edu.uab.ssg.model.snp">SampleBuilder</a></dt>
<dd>
<div class="block">Sets the genotype at a SNP for a particular sample.</div>
</dd>
<dt><a href="edu/uab/ssg/model/snp/Sex.html" title="class in edu.uab.ssg.model.snp"><span class="typeNameLink">Sex</span></a> - Class in <a href="edu/uab/ssg/model/snp/package-summary.html">edu.uab.ssg.model.snp</a></dt>
<dd>
<div class="block">An enum implementation for sex.</div>
</dd>
<dt><a href="edu/uab/ssg/model/snp/SNP.html" title="interface in edu.uab.ssg.model.snp"><span class="typeNameLink">SNP</span></a> - Interface in <a href="edu/uab/ssg/model/snp/package-summary.html">edu.uab.ssg.model.snp</a></dt>
<dd>
<div class="block">A single nucleotide polymorphism (SNP).</div>
</dd>
<dt><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.html" title="class in edu.uab.ssg.io.ucsc"><span class="typeNameLink">SNPAnnotationParser</span></a> - Class in <a href="edu/uab/ssg/io/ucsc/package-summary.html">edu.uab.ssg.io.ucsc</a></dt>
<dd>
<div class="block">A parser for the UCSC SNP annotation format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.html#SNPAnnotationParser--">SNPAnnotationParser()</a></span> - Constructor for class edu.uab.ssg.io.ucsc.<a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.html" title="class in edu.uab.ssg.io.ucsc">SNPAnnotationParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.RecordListener.html" title="interface in edu.uab.ssg.io.ucsc"><span class="typeNameLink">SNPAnnotationParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/ucsc/package-summary.html">edu.uab.ssg.io.ucsc</a></dt>
<dd>
<div class="block">A listener for handling parsed SNP annotation records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/ucsc/SNPAnnotationParser.SNPRecord.html" title="interface in edu.uab.ssg.io.ucsc"><span class="typeNameLink">SNPAnnotationParser.SNPRecord</span></a> - Interface in <a href="edu/uab/ssg/io/ucsc/package-summary.html">edu.uab.ssg.io.ucsc</a></dt>
<dd>
<div class="block">A SNP annotation record.</div>
</dd>
<dt><a href="edu/uab/ssg/io/decode/SNPFileParser.html" title="class in edu.uab.ssg.io.decode"><span class="typeNameLink">SNPFileParser</span></a> - Class in <a href="edu/uab/ssg/io/decode/package-summary.html">edu.uab.ssg.io.decode</a></dt>
<dd>
<div class="block">A parser for the "JHY_SNP.txt" file format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/decode/SNPFileParser.html#SNPFileParser--">SNPFileParser()</a></span> - Constructor for class edu.uab.ssg.io.decode.<a href="edu/uab/ssg/io/decode/SNPFileParser.html" title="class in edu.uab.ssg.io.decode">SNPFileParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/illumina/SNPFileParser.html" title="class in edu.uab.ssg.io.illumina"><span class="typeNameLink">SNPFileParser</span></a> - Class in <a href="edu/uab/ssg/io/illumina/package-summary.html">edu.uab.ssg.io.illumina</a></dt>
<dd>
<div class="block">A parser for the Illumina "SNP_Map.txt" file format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/illumina/SNPFileParser.html#SNPFileParser--">SNPFileParser()</a></span> - Constructor for class edu.uab.ssg.io.illumina.<a href="edu/uab/ssg/io/illumina/SNPFileParser.html" title="class in edu.uab.ssg.io.illumina">SNPFileParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/decode/SNPFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.decode"><span class="typeNameLink">SNPFileParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/decode/package-summary.html">edu.uab.ssg.io.decode</a></dt>
<dd>
<div class="block">A listener for handling parsed SNP records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/illumina/SNPFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.illumina"><span class="typeNameLink">SNPFileParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/illumina/package-summary.html">edu.uab.ssg.io.illumina</a></dt>
<dd>
<div class="block">A listener for handling parsed SNP records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/decode/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.decode"><span class="typeNameLink">SNPFileParser.SNPRecord</span></a> - Interface in <a href="edu/uab/ssg/io/decode/package-summary.html">edu.uab.ssg.io.decode</a></dt>
<dd>
<div class="block">A SNP record.</div>
</dd>
<dt><a href="edu/uab/ssg/io/illumina/SNPFileParser.SNPRecord.html" title="interface in edu.uab.ssg.io.illumina"><span class="typeNameLink">SNPFileParser.SNPRecord</span></a> - Interface in <a href="edu/uab/ssg/io/illumina/package-summary.html">edu.uab.ssg.io.illumina</a></dt>
<dd>
<div class="block">A SNP record.</div>
</dd>
<dt><a href="edu/uab/ssg/model/snp/Strand.html" title="interface in edu.uab.ssg.model.snp"><span class="typeNameLink">Strand</span></a> - Interface in <a href="edu/uab/ssg/model/snp/package-summary.html">edu.uab.ssg.model.snp</a></dt>
<dd>
<div class="block">A strand designation, typically used in the reporting of genotype calls.</div>
</dd>
<dt><a href="edu/uab/ssg/reports/StrandMatchReport.html" title="class in edu.uab.ssg.reports"><span class="typeNameLink">StrandMatchReport</span></a> - Class in <a href="edu/uab/ssg/reports/package-summary.html">edu.uab.ssg.reports</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/reports/StrandMatchReport.html#StrandMatchReport--">StrandMatchReport()</a></span> - Constructor for class edu.uab.ssg.reports.<a href="edu/uab/ssg/reports/StrandMatchReport.html" title="class in edu.uab.ssg.reports">StrandMatchReport</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="I:T">
<!--   -->
</a>
<h2 class="title">T</h2>
<dl>
<dt><a href="edu/uab/ssg/io/omrf/TabParser.html" title="class in edu.uab.ssg.io.omrf"><span class="typeNameLink">TabParser</span></a> - Class in <a href="edu/uab/ssg/io/omrf/package-summary.html">edu.uab.ssg.io.omrf</a></dt>
<dd>
<div class="block">A parser for the tab-delimited text file containing the de-identified and
 coded clinical data from the Oklahoma Medical Research Foundation (OMRF)
 for the 310 samples for which the UAB popgen project has genotypes.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/omrf/TabParser.html#TabParser--">TabParser()</a></span> - Constructor for class edu.uab.ssg.io.omrf.<a href="edu/uab/ssg/io/omrf/TabParser.html" title="class in edu.uab.ssg.io.omrf">TabParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/omrf/TabParser.RecordListener.html" title="interface in edu.uab.ssg.io.omrf"><span class="typeNameLink">TabParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/omrf/package-summary.html">edu.uab.ssg.io.omrf</a></dt>
<dd>
<div class="block">A listener for handling parsed sample records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/omrf/TabParser.SampleRecord.html" title="interface in edu.uab.ssg.io.omrf"><span class="typeNameLink">TabParser.SampleRecord</span></a> - Interface in <a href="edu/uab/ssg/io/omrf/package-summary.html">edu.uab.ssg.io.omrf</a></dt>
<dd>
<div class="block">An sample record.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/IlluminaStrand.html#TOP">TOP</a></span> - Static variable in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/IlluminaStrand.html" title="class in edu.uab.ssg.model.snp">IlluminaStrand</a></dt>
<dd>
<div class="block">The TOP strand.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/LegendFile.html#toString--">toString()</a></span> - Method in class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/LegendFile.html" title="class in edu.uab.ssg.io.marchini_gwas">LegendFile</a></dt>
<dd>
<div class="block">Returns a tabular string representation of this legend.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/AlleleCounter.html#toString--">toString()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/AlleleCounter.html" title="class in edu.uab.ssg.model.snp">AlleleCounter</a></dt>
<dd>
<div class="block">Returns a string representation of this <code>AlleleCounter</code> object.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/DefaultSNP.html#toString--">toString()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/DefaultSNP.html" title="class in edu.uab.ssg.model.snp">DefaultSNP</a></dt>
<dd>
<div class="block">Returns a string representation.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/IlluminaStrand.html#toString--">toString()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/IlluminaStrand.html" title="class in edu.uab.ssg.model.snp">IlluminaStrand</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/model/snp/Sex.html#toString--">toString()</a></span> - Method in class edu.uab.ssg.model.snp.<a href="edu/uab/ssg/model/snp/Sex.html" title="class in edu.uab.ssg.model.snp">Sex</a></dt>
<dd>
<div class="block">Returns a string representation for debugging purposes.</div>
</dd>
</dl>
<a name="I:U">
<!--   -->
</a>
<h2 class="title">U</h2>
<dl>
<dt><a href="edu/uab/ssg/model/snp/Util.html" title="class in edu.uab.ssg.model.snp"><span class="typeNameLink">Util</span></a> - Class in <a href="edu/uab/ssg/model/snp/package-summary.html">edu.uab.ssg.model.snp</a></dt>
<dd>
<div class="block">A collection of utility objects and functions.</div>
</dd>
</dl>
<a name="I:V">
<!--   -->
</a>
<h2 class="title">V</h2>
<dl>
<dt><a href="edu/uab/ssg/io/the1000genomes/VCFParser.html" title="class in edu.uab.ssg.io.the1000genomes"><span class="typeNameLink">VCFParser</span></a> - Class in <a href="edu/uab/ssg/io/the1000genomes/package-summary.html">edu.uab.ssg.io.the1000genomes</a></dt>
<dd>
<div class="block">A parser for the VCF (Variant Call Format) file format described at <a href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-40">http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-40</a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/the1000genomes/VCFParser.html#VCFParser--">VCFParser()</a></span> - Constructor for class edu.uab.ssg.io.the1000genomes.<a href="edu/uab/ssg/io/the1000genomes/VCFParser.html" title="class in edu.uab.ssg.io.the1000genomes">VCFParser</a></dt>
<dd>
<div class="block">Constructs the parser.</div>
</dd>
<dt><a href="edu/uab/ssg/io/the1000genomes/VCFParser.RecordListener.html" title="interface in edu.uab.ssg.io.the1000genomes"><span class="typeNameLink">VCFParser.RecordListener</span></a> - Interface in <a href="edu/uab/ssg/io/the1000genomes/package-summary.html">edu.uab.ssg.io.the1000genomes</a></dt>
<dd>
<div class="block">A listener for handling parsed variant records and problems due to
 bad record formatting.</div>
</dd>
<dt><a href="edu/uab/ssg/io/the1000genomes/VCFParser.VariantRecord.html" title="interface in edu.uab.ssg.io.the1000genomes"><span class="typeNameLink">VCFParser.VariantRecord</span></a> - Interface in <a href="edu/uab/ssg/io/the1000genomes/package-summary.html">edu.uab.ssg.io.the1000genomes</a></dt>
<dd>
<div class="block">A variant record.</div>
</dd>
</dl>
<a name="I:W">
<!--   -->
</a>
<h2 class="title">W</h2>
<dl>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/eigenstrat/GENOWriter.html#write-edu.uab.ssg.model.snp.SNP-java.util.List-">write(SNP, List&lt;Sample&gt;)</a></span> - Method in class edu.uab.ssg.io.eigenstrat.<a href="edu/uab/ssg/io/eigenstrat/GENOWriter.html" title="class in edu.uab.ssg.io.eigenstrat">GENOWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/ent/ENTWriter.html#write-java.util.List-java.util.List-java.io.OutputStream-">write(List&lt;Sample&gt;, List&lt;SNP&gt;, OutputStream)</a></span> - Method in class edu.uab.ssg.io.ent.<a href="edu/uab/ssg/io/ent/ENTWriter.html" title="class in edu.uab.ssg.io.ent">ENTWriter</a></dt>
<dd>
<div class="block">Writes the genotypes for the given samples in a study population 
 to the output stream in the ENT input file format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/fastphase/OUTWriter.html#write-edu.uab.ssg.model.snp.Sample-java.lang.String-">write(Sample, String)</a></span> - Method in class edu.uab.ssg.io.fastphase.<a href="edu/uab/ssg/io/fastphase/OUTWriter.html" title="class in edu.uab.ssg.io.fastphase">OUTWriter</a></dt>
<dd>
<div class="block">Writes the haplotypes for a given sample to the output stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/haploview/MarkerInfoWriter.html#write-java.util.List-java.io.OutputStream-">write(List&lt;SNP&gt;, OutputStream)</a></span> - Method in class edu.uab.ssg.io.haploview.<a href="edu/uab/ssg/io/haploview/MarkerInfoWriter.html" title="class in edu.uab.ssg.io.haploview">MarkerInfoWriter</a></dt>
<dd>
<div class="block">Writes the SNPs to the output stream in the Haploview marker information file format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hyperlasso/DATWriter.html#write-edu.uab.ssg.model.snp.Sample-">write(Sample)</a></span> - Method in class edu.uab.ssg.io.hyperlasso.<a href="edu/uab/ssg/io/hyperlasso/DATWriter.html" title="class in edu.uab.ssg.io.hyperlasso">DATWriter</a></dt>
<dd>
<div class="block">Writes the genotypes for the given sample to the output stream 
 in HyperLasso DAT file format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/GENSWriter.html#write-edu.uab.ssg.model.snp.SNP-">write(SNP)</a></span> - Method in class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/GENSWriter.html" title="class in edu.uab.ssg.io.marchini_gwas">GENSWriter</a></dt>
<dd>
<div class="block">Writes a formatted record for the given SNP to the output stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/HAPSWriter.html#write-java.util.Set-edu.uab.ssg.io.marchini_gwas.Legend-java.io.OutputStream-">write(Set&lt;Sample&gt;, Legend, OutputStream)</a></span> - Method in class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/HAPSWriter.html" title="class in edu.uab.ssg.io.marchini_gwas">HAPSWriter</a></dt>
<dd>
<div class="block">Writes the samples to the output stream in the HAPS file format, recoding alleles to 0 or 1 according to the legend.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/Legend.html#write-java.io.OutputStream-">write(OutputStream)</a></span> - Method in interface edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/Legend.html" title="interface in edu.uab.ssg.io.marchini_gwas">Legend</a></dt>
<dd>
<div class="block">Writes this legend to the given output stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/marchini_gwas/LegendFile.html#write-java.io.OutputStream-">write(OutputStream)</a></span> - Method in class edu.uab.ssg.io.marchini_gwas.<a href="edu/uab/ssg/io/marchini_gwas/LegendFile.html" title="class in edu.uab.ssg.io.marchini_gwas">LegendFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/phase/INPWriter.html#write-java.util.List-java.util.List-java.io.OutputStream-">write(List&lt;Sample&gt;, List&lt;SNP&gt;, OutputStream)</a></span> - Method in class edu.uab.ssg.io.phase.<a href="edu/uab/ssg/io/phase/INPWriter.html" title="class in edu.uab.ssg.io.phase">INPWriter</a></dt>
<dd>
<div class="block">Writes the genotypes for the given samples in a study population 
 to the output stream in PHASE INP format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/MAPWriter.html#write-java.util.List-java.io.OutputStream-">write(List&lt;SNP&gt;, OutputStream)</a></span> - Method in class edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/MAPWriter.html" title="class in edu.uab.ssg.io.plink">MAPWriter</a></dt>
<dd>
<div class="block">Writes the SNPs to the output stream in PLINK MAP file format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/plink/PEDWriter.html#write-edu.uab.ssg.model.snp.Sample-">write(Sample)</a></span> - Method in class edu.uab.ssg.io.plink.<a href="edu/uab/ssg/io/plink/PEDWriter.html" title="class in edu.uab.ssg.io.plink">PEDWriter</a></dt>
<dd>
<div class="block">Writes the genotypes for the given sample to the output stream 
 in PLINK PED format.</div>
</dd>
<dt><span class="memberNameLink"><a href="edu/uab/ssg/io/hyperlasso/DATWriter.html#writeHeader--">writeHeader()</a></span> - Method in class edu.uab.ssg.io.hyperlasso.<a href="edu/uab/ssg/io/hyperlasso/DATWriter.html" title="class in edu.uab.ssg.io.hyperlasso">DATWriter</a></dt>
<dd>
<div class="block">Writes the header (first line, of SNP names).</div>
</dd>
</dl>
<a href="#I:A">A</a>&nbsp;<a href="#I:B">B</a>&nbsp;<a href="#I:C">C</a>&nbsp;<a href="#I:D">D</a>&nbsp;<a href="#I:E">E</a>&nbsp;<a href="#I:F">F</a>&nbsp;<a href="#I:G">G</a>&nbsp;<a href="#I:H">H</a>&nbsp;<a href="#I:I">I</a>&nbsp;<a href="#I:L">L</a>&nbsp;<a href="#I:M">M</a>&nbsp;<a href="#I:O">O</a>&nbsp;<a href="#I:P">P</a>&nbsp;<a href="#I:S">S</a>&nbsp;<a href="#I:T">T</a>&nbsp;<a href="#I:U">U</a>&nbsp;<a href="#I:V">V</a>&nbsp;<a href="#I:W">W</a>&nbsp;</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="overview-summary.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li><a href="overview-tree.html">Tree</a></li>
<li><a href="deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="index.html?index-all.html" target="_top">Frames</a></li>
<li><a href="index-all.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
