<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_101) on Mon Aug 01 11:40:34 CDT 2016 -->
<title>GenotypeFileParser</title>
<meta name="date" content="2016-08-01">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="GenotypeFileParser";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../edu/uab/ssg/io/hapmap/AlleleFreqFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.hapmap"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../../../../edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.hapmap"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?edu/uab/ssg/io/hapmap/GenotypeFileParser.html" target="_top">Frames</a></li>
<li><a href="GenotypeFileParser.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">edu.uab.ssg.io.hapmap</div>
<h2 title="Class GenotypeFileParser" class="title">Class GenotypeFileParser</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://download.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>edu.uab.ssg.io.hapmap.GenotypeFileParser</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public final class <span class="typeNameLabel">GenotypeFileParser</span>
extends <a href="http://download.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></pre>
<div class="block">A parser for the genotype file format for the HapMap phases I+II+III release #28 (NCBI build 36, dbSNP b126) data located at <a href="http://hapmap.ncbi.nlm.nih.gov/downloads/genotypes/2010-08_phaseII+III/">http://hapmap.ncbi.nlm.nih.gov/downloads/genotypes/2010-08_phaseII+III/</a>.
 The file format is described in the README at <a href="http://hapmap.ncbi.nlm.nih.gov/downloads/genotypes/00README.txt">http://hapmap.ncbi.nlm.nih.gov/downloads/genotypes/00README.txt</a>. An excerpt is below:

 <p><tt>
 Col1: refSNP rs# identifier at the time of release (NB might merge with another rs# in the future)<br/>
 Col2: SNP alleles according to dbSNP<br/>
 Col3: chromosome that SNP maps to <br/>
 Col4: chromosome position of SNP, in basepairs on reference sequence<br/>
 Col5: strand of reference sequence that SNP maps to<br/>
 Col6: version of reference sequence assembly<br/>
 Col7: HapMap genotype center that produced the genotypes<br/>
 Col8: LSID for HapMap protocol used for genotyping<br/>
 Col9: LSID for HapMap assay used for genotyping<br/>
 Col10: LSID for panel of individuals genotyped<br/>
 Col11: QC-code, currently 'QC+' for all entries (for future use)<br/>
 Col12 and on: observed genotypes of samples, one per column, sample identifiers in column headers (Coriell catalog numbers, example: NA10847). Duplicate samples have .dup suffix.<br/>
 </tt></p>
 An example, with selected records from genotypes_chr16_CHB_r28_nr.b36_fwd.txt, is below:
 <p><tt>
 rs# alleles chrom pos strand assembly# center protLSID assayLSID panelLSID QCcode NA18524 NA18525 NA18526 NA18527 NA18528 NA18529 NA18531 NA18532 NA18533 NA18534 NA18536 NA18537 NA18538 NA18539 NA18540 NA18541 NA18542 NA18543 NA18544 NA18545 NA18546 NA18547 NA18548 NA18550 NA18552 NA18553 NA18555 NA18557 NA18558 NA18559 NA18560 NA18561 NA18562 NA18563 NA18564 NA18566 NA18567 NA18568 NA18569 NA18570 NA18571 NA18572 NA18573 NA18576 NA18577 NA18579 NA18580 NA18582 NA18583 NA18591 NA18592 NA18593 NA18594 NA18595 NA18596 NA18597 NA18599 NA18602 NA18603 NA18605 NA18608 NA18609 NA18610 NA18611 NA18612 NA18613 NA18614 NA18615 NA18616 NA18617 NA18618 NA18619 NA18620 NA18621 NA18622 NA18623 NA18624 NA18626 NA18627 NA18628 NA18629 NA18630 NA18631 NA18632 NA18633 NA18634 NA18635 NA18636 NA18637 NA18638 NA18639 NA18640 NA18641 NA18642 NA18643 NA18644 NA18645 NA18647 NA18648 NA18649 NA18739 NA18740 NA18741 NA18742 NA18743 NA18745 NA18747 NA18748 NA18749 NA18750 NA18751 NA18752 NA18755 NA18757 NA18758 NA18759 NA18760 NA18761 NA18762 NA18763 NA18765 NA18769 NA18771 NA18772 NA18773 NA18774 NA18777 NA18778 NA18779 NA18780 NA18783 NA18784 NA18785 NA18787 NA18790 NA18792 NA18794 NA18795 NA18798<br/>
 rs3743872 A/G chr16 24045 + ncbi_b36 sanger urn:LSID:illumina.hapmap.org:Protocol:Human_1M_BeadChip:3 urn:LSID:sanger.hapmap.org:Assay:H1Mrs3743872:3 urn:lsid:dcc.hapmap.org:Panel:Han_Chinese:4 QC+ GG GG GG GG GG GG GG AG GG GG GG GG GG GG NN GG GG GG GG GG GG NN GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG AG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG AG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG GG AG GG GG GG GG AG GG AG GG GG GG GG GG GG GG GG GG GG GG GG GG AG GG AG<br/>
 rs2562132 C/T chr16 24170 + ncbi_b36 imsut-riken urn:lsid:imsut-riken.hapmap.org:Protocol:genotyping:1 urn:lsid:imsut-riken.hapmap.org:Assay:2562132:21 urn:LSID:dcc.hapmap.org:Panel:Han_Chinese:1 QC+ CT NN TT NN NN TT NN CT NN NN NN CT NN NN CC NN CT NN NN CC NN CT NN CC CC NN TT NN CT NN NN CC TT TT CC CT NN NN NN TT TT TT TT CT TT CT NN CC NN NN CT CC CT NN NN NN NN NN CC CT CC TT NN TT CC NN NN NN NN NN NN NN CT TT TT CT CT NN NN NN NN NN NN CT CT NN CT TT CT NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN<br/>
 ...<br/>
 rs9673344 C/T chr16 88801983 + ncbi_b36 perlegen urn:lsid:perlegen.hapmap.org:Protocol:Genotyping_1.0.0:2 urn:lsid:perlegen.hapmap.org:Assay:25760.6896899:1 urn:LSID:dcc.hapmap.org:Panel:Han_Chinese:2 QC+ CC NN CC NN NN CC NN CC NN NN NN CC NN NN CC NN CC NN NN CC NN CC NN CC CC NN CC NN CC NN NN CC CC CC CC CC NN NN NN CC CC CC CC CC CC CC NN CC NN NN CC CC CC NN NN NN NN NN CC CC CC CC NN CC CC NN NN NN NN NN NN NN CC CC CC CC CC NN NN NN NN NN NN CC CC NN CC CC CC NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN NN<br/>
 </tt><p>

 The field delimiter is the space character and missing values should be encoded with the 'N' character. Some additional information is available in the README at <a href="http://hapmap.ncbi.nlm.nih.gov/downloads/genotypes/2010-08_phaseII+III/00README.txt">http://hapmap.ncbi.nlm.nih.gov/downloads/genotypes/2010-08_phaseII+III/00README.txt</a>.</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Jelai Wang</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static interface&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.hapmap">GenotypeFileParser.GenotypeRecord</a></span></code>
<div class="block">A genotype record.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static interface&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../edu/uab/ssg/io/hapmap/GenotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.hapmap">GenotypeFileParser.RecordListener</a></span></code>
<div class="block">A listener for handling parsed genotype records and problems due to
 bad record formatting.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../../../../edu/uab/ssg/io/hapmap/GenotypeFileParser.html#GenotypeFileParser--">GenotypeFileParser</a></span>()</code>
<div class="block">Constructs the parser.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../edu/uab/ssg/io/hapmap/GenotypeFileParser.html#parse-java.io.InputStream-edu.uab.ssg.io.hapmap.GenotypeFileParser.RecordListener-">parse</a></span>(<a href="http://download.oracle.com/javase/6/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;in,
     <a href="../../../../../edu/uab/ssg/io/hapmap/GenotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.hapmap">GenotypeFileParser.RecordListener</a>&nbsp;listener)</code>
<div class="block">Parses the input stream for genotype records.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://download.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://download.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#clone--" title="class or interface in java.lang">clone</a>, <a href="http://download.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#equals-java.lang.Object-" title="class or interface in java.lang">equals</a>, <a href="http://download.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#finalize--" title="class or interface in java.lang">finalize</a>, <a href="http://download.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#getClass--" title="class or interface in java.lang">getClass</a>, <a href="http://download.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#hashCode--" title="class or interface in java.lang">hashCode</a>, <a href="http://download.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#notify--" title="class or interface in java.lang">notify</a>, <a href="http://download.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#notifyAll--" title="class or interface in java.lang">notifyAll</a>, <a href="http://download.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#toString--" title="class or interface in java.lang">toString</a>, <a href="http://download.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#wait--" title="class or interface in java.lang">wait</a>, <a href="http://download.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#wait-long-" title="class or interface in java.lang">wait</a>, <a href="http://download.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#wait-long-int-" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="GenotypeFileParser--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>GenotypeFileParser</h4>
<pre>public&nbsp;GenotypeFileParser()</pre>
<div class="block">Constructs the parser.</div>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="parse-java.io.InputStream-edu.uab.ssg.io.hapmap.GenotypeFileParser.RecordListener-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>parse</h4>
<pre>public&nbsp;void&nbsp;parse(<a href="http://download.oracle.com/javase/6/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;in,
                  <a href="../../../../../edu/uab/ssg/io/hapmap/GenotypeFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.hapmap">GenotypeFileParser.RecordListener</a>&nbsp;listener)
           throws <a href="http://download.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Parses the input stream for genotype records.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>in</code> - The input stream, typically a file input stream, of the
 file to be parsed.</dd>
<dd><code>listener</code> - As the input stream is parsed, each genotype record
 is passed to the user-supplied record listener.</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="http://download.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd>
</dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../edu/uab/ssg/io/hapmap/AlleleFreqFileParser.RecordListener.html" title="interface in edu.uab.ssg.io.hapmap"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../../../../edu/uab/ssg/io/hapmap/GenotypeFileParser.GenotypeRecord.html" title="interface in edu.uab.ssg.io.hapmap"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?edu/uab/ssg/io/hapmap/GenotypeFileParser.html" target="_top">Frames</a></li>
<li><a href="GenotypeFileParser.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
